--- ./magic/Magdir/c-lang.ORIGIN	2018-12-13 16:54:22 +0100
+++ ./magic/Magdir/c-lang	2018-12-13 16:58:49 +0100
@@ -58,9 +58,9 @@
 !:strength + 30
 !:mime	text/x-c++
 # But class alone is reduced to avoid beating php (Jens Schleusener)
-0	regex	\^[[:space:]]*class[[:space:]]+[[:digit:][:alpha:]:_]+[[:space:]]*\\{(.*[\n]*)*\\}(;)?$		C++ source text
-!:strength + 13
-!:mime	text/x-c++
+#0	regex	\^[[:space:]]*class[[:space:]]+[[:digit:][:alpha:]:_]+[[:space:]]*\\{(.*[\n]*)*\\}(;)?$		C++ source text
+#!:strength + 13
+#!:mime	text/x-c++
 0	regex	\^[[:space:]]*public:		C++ source text
 !:strength + 30
 !:mime	text/x-c++
