--- magic/Magdir/c-lang.org	2020-04-28 00:40:01 +0000
+++ magic/Magdir/c-lang	2020-04-28 00:42:56 +0000
@@ -73,9 +73,9 @@
 !:mime	text/x-c++
 # But class alone is reduced to avoid beating php (Jens Schleusener)
 0	search/8192	class
->0	regex	\^[[:space:]]*class[[:space:]]+[[:digit:][:alpha:]:_]+[[:space:]]*\\{(.*[\n]*)*\\}(;)?$		C++ source text
-!:strength + 13
-!:mime	text/x-c++
+#>0	regex	\^[[:space:]]*class[[:space:]]+[[:digit:][:alpha:]:_]+[[:space:]]*\\{(.*[\n]*)*\\}(;)?$		C++ source text
+#!:strength + 13
+#!:mime	text/x-c++
 0	search/8192	public
 >0	regex	\^[[:space:]]*public:		C++ source text
 !:strength + 30
