--- ./src/mongo/db/fts/fts_index_format.cpp.ORIGIN	2019-05-17 15:39:49 -0500
+++ ./src/mongo/db/fts/fts_index_format.cpp	2019-05-17 15:40:43 -0500
@@ -241,6 +241,8 @@
             } t;
             uint32_t seed = 0;
             MurmurHash3_x64_128(term.data(), term.size(), seed, t.hash);
+            t.hash[0] = endian::nativeToLittle<uint64_t>(t.hash[0]);
+            t.hash[1] = endian::nativeToLittle<uint64_t>(t.hash[1]);
             string keySuffix = mongo::toHexLower(t.data, sizeof(t.data));
             invariant(termKeySuffixLengthV2 == keySuffix.size());
             b.append("", term.substr(0, termKeyPrefixLengthV2) + keySuffix);
