diff -crN ./GConf/backends/Makefile.am /gestconf/project/GNOME_ACL/GNOME/build/sh_build_GNOME/src/./GConf/backends/Makefile.am
*** ./GConf/backends/Makefile.am	Wed Apr 27 12:08:40 2005
--- /gestconf/project/GNOME_ACL/GNOME/build/sh_build_GNOME/src/./GConf/backends/Makefile.am	Wed Sep 28 08:47:33 2005
***************
*** 19,34 ****
  	xml-entry.c		\
  	xml-backend.c
  
! libgconfbackend_oldxml_la_LDFLAGS = -avoid-version -module -no-undefined
! libgconfbackend_oldxml_la_LIBADD  = $(DEPENDENT_WITH_XML_LIBS) $(top_builddir)/gconf/libgconf-$(MAJOR_VERSION).la $(INTLLIBS)
  
  libgconfbackend_xml_la_SOURCES = 	\
  	markup-backend.c		\
  	markup-tree.h			\
  	markup-tree.c
  
! libgconfbackend_xml_la_LDFLAGS = -avoid-version -module -no-undefined
! libgconfbackend_xml_la_LIBADD  = $(DEPENDENT_LIBS) $(top_builddir)/gconf/libgconf-$(MAJOR_VERSION).la $(INTLLIBS)
  
  noinst_PROGRAMS = xml-test
  
--- 19,34 ----
  	xml-entry.c		\
  	xml-backend.c
  
! libgconfbackend_oldxml_la_LDFLAGS = -avoid-version -module
! libgconfbackend_oldxml_la_LIBADD  = $(DEPENDENT_WITH_XML_LIBS) ../gconf/libgconf-$(MAJOR_VERSION).la $(INTLLIBS)
  
  libgconfbackend_xml_la_SOURCES = 	\
  	markup-backend.c		\
  	markup-tree.h			\
  	markup-tree.c
  
! libgconfbackend_xml_la_LDFLAGS = -avoid-version -module
! libgconfbackend_xml_la_LIBADD  = $(DEPENDENT_LIBS) ../gconf/libgconf-$(MAJOR_VERSION).la $(INTLLIBS)
  
  noinst_PROGRAMS = xml-test
  
diff -crN ./GConf/doc/gconf/Makefile.am /gestconf/project/GNOME_ACL/GNOME/build/sh_build_GNOME/src/./GConf/doc/gconf/Makefile.am
*** ./GConf/doc/gconf/Makefile.am	Tue Sep 14 21:58:09 2004
--- /gestconf/project/GNOME_ACL/GNOME/build/sh_build_GNOME/src/./GConf/doc/gconf/Makefile.am	Fri Nov  5 10:19:46 2004
***************
*** 41,47 ****
  
  # CFLAGS and LDFLAGS for compiling scan program. Only needed
  # if $(DOC_MODULE).types is non-empty.
! GTKDOC_CFLAGS = -I$(top_srcdir) $(CFLAGS) $(DEPENDENT_CFLAGS)
  
  GTKDOC_LIBS = $(LDFLAGS) $(DEPENDENT_LIBS) $(top_builddir)/gconf/libgconf-$(MAJOR_VERSION).la
  
--- 41,48 ----
  
  # CFLAGS and LDFLAGS for compiling scan program. Only needed
  # if $(DOC_MODULE).types is non-empty.
! GTKDOC_CPPFLAGS = -I$(top_srcdir) @CPPFLAGS@
! GTKDOC_CFLAGS = $(DEPENDENT_CFLAGS)
  
  GTKDOC_LIBS = $(LDFLAGS) $(DEPENDENT_LIBS) $(top_builddir)/gconf/libgconf-$(MAJOR_VERSION).la
  
diff -crN ./GConf/tests/Makefile.am /gestconf/project/GNOME_ACL/GNOME/build/sh_build_GNOME/src/./GConf/tests/Makefile.am
*** ./GConf/tests/Makefile.am	Thu Sep 19 04:49:09 2002
--- /gestconf/project/GNOME_ACL/GNOME/build/sh_build_GNOME/src/./GConf/tests/Makefile.am	Fri Jun 27 11:15:48 2003
***************
*** 2,8 ****
  EFENCE=
  
  INCLUDES = -I$(top_srcdir) -I$(top_builddir) \
! 	 $(DEPENDENT_CFLAGS) \
  	 -DG_LOG_DOMAIN=\"GConf-Tests\" -DGCONF_ENABLE_INTERNALS=1
  
  noinst_PROGRAMS=testgconf testlisteners testschemas testchangeset testencode testunique testpersistence testdirlist testaddress testbackend
--- 2,8 ----
  EFENCE=
  
  INCLUDES = -I$(top_srcdir) -I$(top_builddir) \
! 	 $(DEPENDENT_CFLAGS) -I$(top_builddir)/gconf \
  	 -DG_LOG_DOMAIN=\"GConf-Tests\" -DGCONF_ENABLE_INTERNALS=1
  
  noinst_PROGRAMS=testgconf testlisteners testschemas testchangeset testencode testunique testpersistence testdirlist testaddress testbackend
