Index: db-4.8.24/dist/Makefile.in
===================================================================
--- db-4.8.24.orig/dist/Makefile.in	2009-08-17 15:26:18.000000000 +0200
+++ db-4.8.24/dist/Makefile.in	2010-01-18 19:10:31.000000000 +0100
@@ -958,7 +958,8 @@
 	    ($(MKDIR) -p $(DESTDIR)$(libdir) && \
 	    $(CHMOD) $(dmode) $(DESTDIR)$(libdir))
 	@cd $(DESTDIR)$(libdir) && $(RM) $(LIB_INSTALL_FILE_LIST)
-	@$(INSTALLER) @INSTALL_LIBS@ $(DESTDIR)$(libdir)
+	$(INSTALLER) @INSTALL_LIBS@ $(DESTDIR)$(libdir)
+	$(INSTALLER) .libs/$(libdb_version) $(DESTDIR)$(libdir)
 	@(cd $(DESTDIR)$(libdir) && \
 	    test -f $(libso) && $(LN) -s $(libso) $(libso_default); \
 	    test -f $(libso) && $(LN) -s $(libso) $(libso_major); \
