diff -crN ./docbookx/LICENSE /gestconf/project/GNOME_ACL/GNOME/build/sh_build_GNOME/src/./docbookx/LICENSE
*** ./docbookx/LICENSE	Thu Jan  1 01:00:00 1970
--- /gestconf/project/GNOME_ACL/GNOME/build/sh_build_GNOME/src/./docbookx/LICENSE	Tue Aug 12 10:26:42 2003
***************
*** 0 ****
--- 1,302 ----
+ 	
+ International License Agreement for Non-Warranted Programs
+ 
+ 
+ Part 1 - General Terms
+ 
+ 
+ PLEASE READ THIS AGREEMENT CAREFULLY BEFORE USING THE PROGRAM. IBM WILL LICENSE
+ THE PROGRAM TO YOU ONLY IF YOU FIRST ACCEPT THE TERMS OF THIS AGREEMENT. BY
+ USING THE PROGRAM YOU AGREE TO THESE TERMS. IF YOU DO NOT AGREE TO THE TERMS OF
+ THIS AGREEMENT, PROMPTLY RETURN THE UNUSED PROGRAM TO THE PARTY (EITHER IBM OR
+ ITS RESELLER) FROM WHOM YOU ACQUIRED IT TO RECEIVE A REFUND OF THE AMOUNT YOU
+ PAID (IF ANY).
+ 
+ The Program is copyrighted and licensed, not sold.
+ 
+ The term "Program" means the original program and all whole or partial copies
+ of it. A Program consists of machine-readable instructions, its components,
+ data, audio-visual content (such as images, text, recordings, or pictures), and
+ related licensed materials.
+ 
+ This Agreement includes Part 1 - General Terms and Part 2 - Country-unique
+ Terms and is the complete agreement regarding the use of this Program, and
+ replaces any prior oral or written communications between you and IBM. The
+ terms of Part 2 may replace or modify those of Part 1.
+ 
+ 
+ 1. License
+ 
+ Use of the Program
+ 
+ IBM grants you a nonexclusive license to use the Program.
+ 
+ You may 1) use the Program to the extent of authorizations you have acquired
+ and 2) make and install copies to support the level of use authorized,
+ providing you reproduce the copyright notice and any other legends of ownership
+ on each copy, or partial copy, of the Program.
+ 
+ If you acquire this Program as a program upgrade, your authorization to use the
+ Program from which you upgraded is terminated.
+ 
+ You will ensure that anyone who uses the Program does so only in compliance
+ with the terms of this Agreement.
+ 
+ You may not 1) use, copy, modify, or distribute the Program except as provided
+ in this Agreement; 2) reverse assemble, reverse compile, or otherwise translate
+ the Program except as specifically permitted by law without the possibility of
+ contractual waiver; or 3) sublicense, rent, or lease the Program.
+ 
+ Transfer of Rights and Obligations
+ 
+ You may transfer all your license rights and obligations under a Proof of
+ Entitlement for the Program to another party by transferring the Proof of
+ Entitlement and a copy of this Agreement and all documentation. The transfer of
+ your license rights and obligations terminates your authorization to use the
+ Program under the Proof of Entitlement.
+ 
+ 
+ 2. Proof of Entitlement
+ 
+ The Proof of Entitlement for this Program is evidence of your authorization to
+ use this Program and of your eligibility for future upgrade program prices (if
+ announced) and potential special or promotional opportunities.
+ 
+ 
+ 3. Charges and Taxes
+ 
+ IBM defines use for the Program for charging purposes and specifies it in the
+ Proof of Entitlement. Charges are based on extent of use authorized. If you
+ wish to increase the extent of use, notify IBM or its reseller and pay any
+ applicable charges. IBM does not give refunds or credits for charges already
+ due or paid.
+ 
+ If any authority imposes a duty, tax, levy or fee, excluding those based on
+ IBM's net income, upon the Program supplied by IBM under this Agreement, then
+ you agree to pay that amount as IBM specifies or supply exemption documentation.
+ 
+ 
+ 4. No Warranty
+ 
+ SUBJECT TO ANY STATUTORY WARRANTIES WHICH CAN NOT BE EXCLUDED, IBM MAKES NO
+ WARRANTIES OR CONDITIONS EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT
+ LIMITATION, THE WARRANTY OF NON-INFRINGEMENT AND THE IMPLIED WARRANTIES OF
+ MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE, REGARDING THE PROGRAM OR
+ TECHNICAL SUPPORT, IF ANY. IBM MAKES NO WARRANTY REGARDING THE CAPABILITY OF
+ THE PROGRAM TO CORRECTLY PROCESS, PROVIDE AND/OR RECEIVE DATE DATA WITHIN AND
+ BETWEEN THE 20TH AND 21ST CENTURIES.
+ 
+ The exclusion also applies to any of IBM's subcontractors, suppliers, or
+ program developers (collectively called "Suppliers").
+ 
+ Manufacturers, suppliers, or publishers of non-IBM Programs may provide their
+ own warranties.
+ 
+ 
+ 5. Limitation of Liability
+ 
+ NEITHER IBM NOR ITS SUPPLIERS WILL BE LIABLE FOR ANY DIRECT OR INDIRECT
+ DAMAGES, INCLUDING WITHOUT LIMITATION, LOST PROFITS, LOST SAVINGS, OR ANY
+ INCIDENTAL, SPECIAL, OR OTHER ECONOMIC CONSEQUENTIAL DAMAGES, EVEN IF IBM IS
+ INFORMED OF THEIR POSSIBILITY. SOME JURISDICTIONS DO NOT ALLOW THE EXCLUSION OR
+ LIMITATION OF INCIDENTAL OR CONSEQUENTIAL DAMAGES, SO THE ABOVE EXCLUSION OR
+ LIMITATION MAY NOT APPLY TO YOU.
+ 
+ 
+ 6. General
+ 
+ Nothing in this Agreement affects any statutory rights of consumers that cannot
+ be waived or limited by contract.
+ 
+ IBM may terminate your license if you fail to comply with the terms of this
+ Agreement. If IBM does so, you must immediately destroy the Program and all
+ copies you made of it.
+ 
+ You agree to comply with applicable export laws and regulations.
+ 
+ Neither you nor IBM will bring a legal action under this Agreement more than
+ two years after the cause of action arose unless otherwise provided by local
+ law without the possibility of contractual waiver or limitation.
+ 
+ Neither you nor IBM is responsible for failure to fulfill any obligations due
+ to causes beyond its control.
+ 
+ IBM does not provide program services or technical support, unless IBM
+ specifies otherwise.
+ 
+ The laws of the country in which you acquire the Program govern this Agreement,
+ except 1) in Australia, the laws of the State or Territory in which the
+ transaction is performed govern this Agreement; 2) in Albania, Armenia,
+ Belarus, Bosnia/Herzegovina, Bulgaria, Croatia, Czech Republic, Georgia,
+ Hungary, Kazakhstan, Kirghizia, Former Yugoslav Republic of Macedonia (FYROM),
+ Moldova, Poland, Romania, Russia, Slovak Republic, Slovenia, Ukraine, and
+ Federal Republic of Yugoslavia, the laws of Austria govern this Agreement;
+ 3) in the United Kingdom, all disputes relating to this Agreement will be
+ governed by English Law and will be submitted to the exclusive jurisdiction of
+ the English courts; 4) in Canada, the laws in the Province of Ontario govern
+ this Agreement; and 5) in the United States and Puerto Rico, and People's
+ Republic of China, the laws of the State of New York govern this Agreement.
+ 
+ 
+ 
+ Part 2 - Country-unique Terms
+ 
+ 
+ AUSTRALIA:
+ 
+ No Warranty (Section 4):
+ 
+ The following paragraph is added to this Section:
+ 
+ Although IBM specifies that there are no warranties, you may have certain
+ rights under the Trade Practices Act 1974 or other legislation and are only
+ limited to the extent permitted by the applicable legislation.
+ 
+ Limitation of Liability (Section 5):
+ 
+ The following paragraph is added to this Section:
+ 
+ Where IBM is in breach of a condition or warranty implied by the Trade
+ Practices Act 1974, IBM's liability is limited to the repair or replacement of
+ the goods, or the supply of equivalent goods. Where that condition or warranty
+ relates to right to sell, quiet possession or clear title, or the goods are of
+ a kind ordinarily acquired for personal, domestic or household use or
+ consumption, then none of the limitations in this paragraph apply.
+ 
+ 
+ GERMANY:
+ 
+ No Warranty (Section 4):
+ 
+ The following paragraphs are added to this Section:
+ 
+ The minimum warranty period for Programs is six months.
+ 
+ In case a Program is delivered without Specifications, we will only warrant
+ that the Program information correctly describes the Program and that the
+ Program can be used according to the Program information. You have to check the
+ usability according to the Program information within the "money-back guaranty"
+ period.
+ 
+ Limitation of Liability (Section 5):
+ 
+ The following paragraph is added to this Section:
+ 
+ The limitations and exclusions specified in the Agreement will not apply to
+ damages caused by IBM with fraud or gross negligence, and for express warranty.
+ 
+ 
+ INDIA:
+ 
+ General (Section 6):
+ 
+ The following replaces the fourth paragraph of this Section:
+ 
+ If no suit or other legal action is brought, within two years after the cause
+ of action arose, in respect of any claim that either party may have against the
+ other, the rights of the concerned party in respect of such claim will be
+ forfeited and the other party will stand released from its obligations in
+ respect of such claim.
+ 
+ 
+ IRELAND:
+ 
+ No Warranty (Section 4):
+ 
+ The following paragraph is added to this Section:
+ 
+ Except as expressly provided in these terms and conditions, all statutory
+ conditions, including all warranties implied, but without prejudice to the
+ generality of the foregoing, all warranties implied by the Sale of Goods Act
+ 1893 or the Sale of Goods and Supply of Services Act 1980 are hereby excluded.
+ 
+ 
+ ITALY:
+ 
+ Limitation of Liability (Section 5):
+ 
+ This Section is replaced by the following:
+ 
+ Unless otherwise provided by mandatory law, IBM is not liable for any damages
+ which might arise.
+ 
+ 
+ NEW ZEALAND:
+ 
+ No Warranty (Section 4):
+ 
+ The following paragraph is added to this Section:
+ 
+ Although IBM specifies that there are no warranties, you may have certain
+ rights under the Consumer Guarantees Act 1993 or other legislation which cannot
+ be excluded or limited. The Consumer Guarantees Act 1993 will not apply in
+ respect of any goods or services which IBM provides, if you require the goods
+ and services for the purposes of a business as defined in that Act.
+ 
+ Limitation of Liability (Section 5):
+ 
+ The following paragraph is added to this Section:
+ 
+ Where Programs are not acquired for the purposes of a business as defined in
+ the Consumer Guarantees Act 1993, the limitations in this Section are subject
+ to the limitations in that Act.
+ 
+ 
+ PEOPLE'S REPUBLIC OF CHINA:
+ 
+ Charges (Section 3):
+ 
+ The following paragraph is added to the Section:
+ 
+ All banking charges incurred in the People's Republic of China will be borne by
+ you and those incurred outside the People's Republic of China will be borne by
+ IBM.
+ 
+ 
+ UNITED KINGDOM:
+ 
+ Limitation of Liability (Section 5):
+ 
+ The following paragraph is added to this Section at the end of the first
+ paragraph:
+ 
+ The limitation of liability will not apply to any breach of IBM's obligations
+ implied by Section 12 of the Sale of Goods Act 1979 or Section 2 of the Supply
+ of Goods and Services Act 1982.
+ 
+ 
+ 
+ 
+ 
+ 
+ If available, the original license text for this package will follow below.
+ ===============================================================================
+ ===============================================================================
+ <!-- ...................................................................... -->
+ <!-- DocBook XML DTD V4.1.2 ................................................. -->
+ <!-- File docbookx.dtd .................................................... -->
+ 
+ <!-- Copyright 1992-2000 HaL Computer Systems, Inc.,
+      O'Reilly & Associates, Inc., ArborText, Inc., Fujitsu Software
+      Corporation, Norman Walsh, and the Organization for the Advancement
+      of Structured Information Standards (OASIS).
+ 
+      $Id: LICENSE,v 1.1.2.2 2003/08/12 09:26:41 bergerv Exp $
+ 
+      Permission to use, copy, modify and distribute the DocBook XML DTD
+      and its accompanying documentation for any purpose and without fee
+      is hereby granted in perpetuity, provided that the above copyright
+      notice and this paragraph appear in all copies.  The copyright
+      holders make no representation about the suitability of the DTD for
+      any purpose.  It is provided "as is" without expressed or implied
+      warranty.
+ 
+      If you modify the DocBook DTD in any way, except for declaring and
+      referencing additional sets of general entities and declaring
+      additional notations, label your DTD as a variant of DocBook.  See
+      the maintenance documentation for more information.
+ 
+      Please direct all questions, bug reports, or suggestions for 
+      changes to the docbook@lists.oasis-open.org mailing list. For more
+      information, see http://www.oasis-open.org/docbook/.
+ -->
diff -crN ./docbookx/Makefile /gestconf/project/GNOME_ACL/GNOME/build/sh_build_GNOME/src/./docbookx/Makefile
*** ./docbookx/Makefile	Sat Feb 16 20:08:29 2002
--- /gestconf/project/GNOME_ACL/GNOME/build/sh_build_GNOME/src/./docbookx/Makefile	Thu Dec  5 15:57:00 2002
***************
*** 1,7 ****
  SHELL=/bin/sh
! VERSION=4.1.2-9
! DESTDIR=/usr/share/xml/docbook/xml-dtd-$(VERSION)
! CATALOG=/usr/share/xml/docbook/catalog
  OPENJADEVER=1.3
  
  all:
--- 1,8 ----
  SHELL=/bin/sh
! VERSION=4.1.2
! INSTALL_PATH ?= /usr
! DESTDIR=$(INSTALL_PATH)/share/xml/docbook/xml-dtd-$(VERSION)
! CATALOG=$(INSTALL_PATH)/share/xml/docbook/catalog
  OPENJADEVER=1.3
  
  all:
***************
*** 10,21 ****
  
  install-dtd:
  	mkdir -p $(DESTDIR)/ent	
! 	chown -R root.root .
! 	chmod -R a+rX,g-w,o-w .
! 	install docbook.cat $(DESTDIR)/catalog
! 	install *.dtd $(DESTDIR)
! 	install *.mod $(DESTDIR)
! 	install ent/* $(DESTDIR)/ent
  
  
  register-catalog:
--- 11,22 ----
  
  install-dtd:
  	mkdir -p $(DESTDIR)/ent	
! 	#chown -R root.root .
! 	#chmod -R a+rX,g-w,o-w .
! 	cp docbook.cat $(DESTDIR)/catalog
! 	cp *.dtd $(DESTDIR)
! 	cp *.mod $(DESTDIR)
! 	cp ent/* $(DESTDIR)/ent
  
  
  register-catalog:
***************
*** 24,30 ****
  	#
          #
  	# DocBook XML V4.1.2
! 	@./buildDocBookCatalog
  
  uninstall:
  	#
--- 25,31 ----
  	#
          #
  	# DocBook XML V4.1.2
! 	@$(SHELL) ./buildDocBookCatalog
  
  uninstall:
  	#
***************
*** 32,35 ****
  	#
  	#
  	# DocBook XML V4.1.2
! 	@./deleteDocBookCatalog $(DESTDIR)
--- 33,36 ----
  	#
  	#
  	# DocBook XML V4.1.2
! 	@$(SHELL) ./deleteDocBookCatalog $(DESTDIR)
diff -crN ./docbookx/buildDocBookCatalog /gestconf/project/GNOME_ACL/GNOME/build/sh_build_GNOME/src/./docbookx/buildDocBookCatalog
*** ./docbookx/buildDocBookCatalog	Sun Feb 17 18:32:45 2002
--- /gestconf/project/GNOME_ACL/GNOME/build/sh_build_GNOME/src/./docbookx/buildDocBookCatalog	Fri Oct  4 10:14:15 2002
***************
*** 28,45 ****
  # Modifications made to Daniel Veillard's catalog build script at 
  # http://xmlsoft.org/buildDocBookCatalog.
  
! if [ ! -d /etc/xml -a -w /etc ]
  then
!     echo Directory /etc/xml missing creating it
!     mkdir /etc/xml
!     chmod 755 /etc/xml
  fi
! if [ -w /etc/xml ]
  then
!     ROOTCATALOG=/etc/xml/catalog
!     CATALOG=/etc/xml/docbook
  else
!     echo Unable to update root catalog /etc/xml/catalog
      echo Using $HOME/xmlcatalog as the root catalog
      echo export XMLCATALOG=$HOME/xmlcatalog
      ROOTCATALOG=$HOME/xmlcatalog
--- 28,62 ----
  # Modifications made to Daniel Veillard's catalog build script at 
  # http://xmlsoft.org/buildDocBookCatalog.
  
! XMLCATALOG='xmlcatalog'
! 
! if [ "$DATADIR" = "" ]
! then
! 	if [ "$INSTALL_PATH" = "" ]
! 	then
! 		DATADIR=/usr/share/xml
! 	else
! 		DATADIR=$INSTALL_PATH/share/xml
! 	fi
! fi
! 
! if [ "$SYSCONFDIR" = "" ]
! then
! 	SYSCONFDIR=$INSTALL_PATH/etc/xml
! fi
! 
! if [ ! -d "$SYSCONFDIR" ]
  then
!     echo Directory $SYSCONFDIR missing creating it
!     mkdir -p $SYSCONFDIR
!     chmod 755 $SYSCONFDIR
  fi
! if [ -w "$SYSCONFDIR" ]
  then
!     ROOTCATALOG="$SYSCONFDIR/catalog"
!     CATALOG="$SYSCONFDIR/docbook"
  else
!     echo Unable to update root catalog $SYSCONFDIR/catalog
      echo Using $HOME/xmlcatalog as the root catalog
      echo export XMLCATALOG=$HOME/xmlcatalog
      ROOTCATALOG=$HOME/xmlcatalog
***************
*** 51,57 ****
  #
  if [ ! -r $ROOTCATALOG ] ; then
     echo creating XML Catalog root $ROOTCATALOG
!    /usr/bin/xmlcatalog --noout --create $ROOTCATALOG
  fi
  if [ ! -r $ROOTCATALOG ] ; then
     echo Failed creating XML Catalog root $ROOTCATALOG
--- 68,74 ----
  #
  if [ ! -r $ROOTCATALOG ] ; then
     echo creating XML Catalog root $ROOTCATALOG
!    ${XMLCATALOG} --noout --create $ROOTCATALOG
  fi
  if [ ! -r $ROOTCATALOG ] ; then
     echo Failed creating XML Catalog root $ROOTCATALOG
***************
*** 59,65 ****
  fi
  if [ ! -r $CATALOG ] ; then
     echo creating DocBook XML Catalog $CATALOG
!    /usr/bin/xmlcatalog --noout --create $CATALOG
  fi
  if [ ! -r $CATALOG ] ; then
     echo Failed creating DocBook XML Catalog $CATALOG
--- 76,82 ----
  fi
  if [ ! -r $CATALOG ] ; then
     echo creating DocBook XML Catalog $CATALOG
!    ${XMLCATALOG} --noout --create $CATALOG
  fi
  if [ ! -r $CATALOG ] ; then
     echo Failed creating DocBook XML Catalog $CATALOG
***************
*** 69,75 ****
  #
  # find the prefix for DocBook DTD
  #
! found=`find /usr/share/xml -name docbookx.dtd`
  if [ "$found" = "" ] ; then
      found=`find $HOME -name docbookx.dtd`
  fi
--- 86,92 ----
  #
  # find the prefix for DocBook DTD
  #
! found=`find ${DATADIR} -name docbookx.dtd`
  if [ "$found" = "" ] ; then
      found=`find $HOME -name docbookx.dtd`
  fi
***************
*** 105,151 ****
  #    exit 2
  #fi
  
! xmlcatalog --noout --add "public" \
      "-//OASIS//ELEMENTS DocBook XML Information Pool V4.1.2//EN" \
      "file://$docbookdir/dbpoolx.mod" $CATALOG
! xmlcatalog --noout --add "public" \
      "-//OASIS//DTD DocBook XML V4.1.2//EN" \
      "file://$docbookdir/docbookx.dtd" $CATALOG
! xmlcatalog --noout --add "public" \
      "-//OASIS//ENTITIES DocBook XML Character Entities V4.1.2//EN" \
      "file://$docbookdir/dbcentx.mod" $CATALOG
! xmlcatalog --noout --add "public" \
      "-//OASIS//ENTITIES DocBook XML Notations V4.1.2//EN" \
      "file://$docbookdir/dbnotnx.mod" $CATALOG
! xmlcatalog --noout --add "public" \
      "-//OASIS//ENTITIES DocBook XML Additional General Entities V4.1.2//EN" \
      "file://$docbookdir/dbgenent.mod" $CATALOG
! xmlcatalog --noout --add "public" \
      "-//OASIS//ELEMENTS DocBook XML Document Hierarchy V4.1.2//EN" \
      "file://$docbookdir/dbhierx.mod" $CATALOG
! xmlcatalog --noout --add "public" \
      "-//OASIS//DTD XML Exchange Table Model 19990315//EN" \
      "file://$docbookdir/soextblx.dtd" $CATALOG
! xmlcatalog --noout --add "public" \
      "-//OASIS//DTD DocBook XML CALS Table Model V4.1.2//EN" \
      "file://$docbookdir/calstblx.dtd" $CATALOG
! xmlcatalog --noout --add "rewriteSystem" \
      "http://www.oasis-open.org/docbook/xml/4.1.2" \
      "file://$docbookdir" $CATALOG
! xmlcatalog --noout --add "rewriteURI" \
      "http://www.oasis-open.org/docbook/xml/4.1.2" \
      "file://$docbookdir" $CATALOG
  
! xmlcatalog --noout --add "delegatePublic" \
      "-//OASIS//ENTITIES DocBook XML" \
      "file://$CATALOG" $ROOTCATALOG
! xmlcatalog --noout --add "delegatePublic" \
      "-//OASIS//DTD DocBook XML" \
      "file://$CATALOG" $ROOTCATALOG
! xmlcatalog --noout --add "delegateSystem" \
      "http://www.oasis-open.org/docbook/" \
      "file://$CATALOG" $ROOTCATALOG
! xmlcatalog --noout --add "delegateURI" \
      "http://www.oasis-open.org/docbook/" \
      "file://$CATALOG" $ROOTCATALOG
  
--- 122,168 ----
  #    exit 2
  #fi
  
! ${XMLCATALOG} --noout --add "public" \
      "-//OASIS//ELEMENTS DocBook XML Information Pool V4.1.2//EN" \
      "file://$docbookdir/dbpoolx.mod" $CATALOG
! ${XMLCATALOG} --noout --add "public" \
      "-//OASIS//DTD DocBook XML V4.1.2//EN" \
      "file://$docbookdir/docbookx.dtd" $CATALOG
! ${XMLCATALOG} --noout --add "public" \
      "-//OASIS//ENTITIES DocBook XML Character Entities V4.1.2//EN" \
      "file://$docbookdir/dbcentx.mod" $CATALOG
! ${XMLCATALOG} --noout --add "public" \
      "-//OASIS//ENTITIES DocBook XML Notations V4.1.2//EN" \
      "file://$docbookdir/dbnotnx.mod" $CATALOG
! ${XMLCATALOG} --noout --add "public" \
      "-//OASIS//ENTITIES DocBook XML Additional General Entities V4.1.2//EN" \
      "file://$docbookdir/dbgenent.mod" $CATALOG
! ${XMLCATALOG} --noout --add "public" \
      "-//OASIS//ELEMENTS DocBook XML Document Hierarchy V4.1.2//EN" \
      "file://$docbookdir/dbhierx.mod" $CATALOG
! ${XMLCATALOG} --noout --add "public" \
      "-//OASIS//DTD XML Exchange Table Model 19990315//EN" \
      "file://$docbookdir/soextblx.dtd" $CATALOG
! ${XMLCATALOG} --noout --add "public" \
      "-//OASIS//DTD DocBook XML CALS Table Model V4.1.2//EN" \
      "file://$docbookdir/calstblx.dtd" $CATALOG
! ${XMLCATALOG} --noout --add "rewriteSystem" \
      "http://www.oasis-open.org/docbook/xml/4.1.2" \
      "file://$docbookdir" $CATALOG
! ${XMLCATALOG} --noout --add "rewriteURI" \
      "http://www.oasis-open.org/docbook/xml/4.1.2" \
      "file://$docbookdir" $CATALOG
  
! ${XMLCATALOG} --noout --add "delegatePublic" \
      "-//OASIS//ENTITIES DocBook XML" \
      "file://$CATALOG" $ROOTCATALOG
! ${XMLCATALOG} --noout --add "delegatePublic" \
      "-//OASIS//DTD DocBook XML" \
      "file://$CATALOG" $ROOTCATALOG
! ${XMLCATALOG} --noout --add "delegateSystem" \
      "http://www.oasis-open.org/docbook/" \
      "file://$CATALOG" $ROOTCATALOG
! ${XMLCATALOG} --noout --add "delegateURI" \
      "http://www.oasis-open.org/docbook/" \
      "file://$CATALOG" $ROOTCATALOG
  
***************
*** 155,161 ****
  top=`dirname $docbookdir`
  found=`find $top -name iso-amsb.ent`
  if [ "$found" = "" ] ; then
!     found=`find /usr/share/xml -name iso-amsb.ent`
  fi
  if [ "$found" = "" ] ; then
      found=`find $HOME -name iso-amsb.ent`
--- 172,178 ----
  top=`dirname $docbookdir`
  found=`find $top -name iso-amsb.ent`
  if [ "$found" = "" ] ; then
!     found=`find ${DATADIR} -name iso-amsb.ent`
  fi
  if [ "$found" = "" ] ; then
      found=`find $HOME -name iso-amsb.ent`
***************
*** 187,251 ****
  isodir=`dirname $entxml`
  echo Found ISO DocBook entities in $isodir
  
! xmlcatalog --noout --add "public" \
      "ISO 8879:1986//ENTITIES Publishing//EN" \
      "file://$isodir/iso-pub.ent" $CATALOG
! xmlcatalog --noout --add "public" \
      "ISO 8879:1986//ENTITIES Greek Letters//EN" \
      "file://$isodir/iso-grk1.ent" $CATALOG
! xmlcatalog --noout --add "public" \
      "ISO 8879:1986//ENTITIES Box and Line Drawing//EN" \
      "file://$isodir/iso-box.ent" $CATALOG
! xmlcatalog --noout --add "public" \
      "ISO 8879:1986//ENTITIES Greek Symbols//EN" \
      "file://$isodir/iso-grk3.ent" $CATALOG
! xmlcatalog --noout --add "public" \
      "ISO 8879:1986//ENTITIES Added Math Symbols: Negated Relations//EN" \
      "file://$isodir/iso-amsn.ent" $CATALOG
! xmlcatalog --noout --add "public" \
      "ISO 8879:1986//ENTITIES Numeric and Special Graphic//EN" \
      "file://$isodir/iso-num.ent" $CATALOG
! xmlcatalog --noout --add "public" \
      "ISO 8879:1986//ENTITIES Alternative Greek Symbols//EN" \
      "file://$isodir/iso-grk4.ent" $CATALOG
! xmlcatalog --noout --add "public" \
      "ISO 8879:1986//ENTITIES Diacritical Marks//EN" \
      "file://$isodir/iso-dia.ent" $CATALOG
! xmlcatalog --noout --add "public" \
      "ISO 8879:1986//ENTITIES Monotoniko Greek//EN" \
      "file://$isodir/iso-grk2.ent" $CATALOG
! xmlcatalog --noout --add "public" \
      "ISO 8879:1986//ENTITIES Added Math Symbols: Arrow Relations//EN" \
      "file://$isodir/iso-amsa.ent" $CATALOG
! xmlcatalog --noout --add "public" \
      "ISO 8879:1986//ENTITIES Added Math Symbols: Ordinary//EN" \
      "file://$isodir/iso-amso.ent" $CATALOG
! xmlcatalog --noout --add "public" \
      "ISO 8879:1986//ENTITIES Russian Cyrillic//EN" \
      "file://$isodir/iso-cyr1.ent" $CATALOG
! xmlcatalog --noout --add "public" \
      "ISO 8879:1986//ENTITIES General Technical//EN" \
      "file://$isodir/iso-tech.ent" $CATALOG
! xmlcatalog --noout --add "public" \
      "ISO 8879:1986//ENTITIES Added Math Symbols: Delimiters//EN" \
      "file://$isodir/iso-amsc.ent" $CATALOG
! xmlcatalog --noout --add "public" \
      "ISO 8879:1986//ENTITIES Added Latin 1//EN" \
      "file://$isodir/iso-lat1.ent" $CATALOG
! xmlcatalog --noout --add "public" \
      "ISO 8879:1986//ENTITIES Added Math Symbols: Binary Operators//EN" \
      "file://$isodir/iso-amsb.ent" $CATALOG
! xmlcatalog --noout --add "public" \
      "ISO 8879:1986//ENTITIES Added Latin 2//EN" \
      "file://$isodir/iso-lat2.ent" $CATALOG
! xmlcatalog --noout --add "public" \
      "ISO 8879:1986//ENTITIES Added Math Symbols: Relations//EN" \
      "file://$isodir/iso-amsr.ent" $CATALOG
! xmlcatalog --noout --add "public" \
      "ISO 8879:1986//ENTITIES Non-Russian Cyrillic//EN" \
      "file://$isodir/iso-cyr2.ent" $CATALOG
  
! xmlcatalog --noout --add "delegatePublic" \
      "ISO 8879:1986" \
      "file://$CATALOG" $ROOTCATALOG
  #
--- 204,268 ----
  isodir=`dirname $entxml`
  echo Found ISO DocBook entities in $isodir
  
! ${XMLCATALOG} --noout --add "public" \
      "ISO 8879:1986//ENTITIES Publishing//EN" \
      "file://$isodir/iso-pub.ent" $CATALOG
! ${XMLCATALOG} --noout --add "public" \
      "ISO 8879:1986//ENTITIES Greek Letters//EN" \
      "file://$isodir/iso-grk1.ent" $CATALOG
! ${XMLCATALOG} --noout --add "public" \
      "ISO 8879:1986//ENTITIES Box and Line Drawing//EN" \
      "file://$isodir/iso-box.ent" $CATALOG
! ${XMLCATALOG} --noout --add "public" \
      "ISO 8879:1986//ENTITIES Greek Symbols//EN" \
      "file://$isodir/iso-grk3.ent" $CATALOG
! ${XMLCATALOG} --noout --add "public" \
      "ISO 8879:1986//ENTITIES Added Math Symbols: Negated Relations//EN" \
      "file://$isodir/iso-amsn.ent" $CATALOG
! ${XMLCATALOG} --noout --add "public" \
      "ISO 8879:1986//ENTITIES Numeric and Special Graphic//EN" \
      "file://$isodir/iso-num.ent" $CATALOG
! ${XMLCATALOG} --noout --add "public" \
      "ISO 8879:1986//ENTITIES Alternative Greek Symbols//EN" \
      "file://$isodir/iso-grk4.ent" $CATALOG
! ${XMLCATALOG} --noout --add "public" \
      "ISO 8879:1986//ENTITIES Diacritical Marks//EN" \
      "file://$isodir/iso-dia.ent" $CATALOG
! ${XMLCATALOG} --noout --add "public" \
      "ISO 8879:1986//ENTITIES Monotoniko Greek//EN" \
      "file://$isodir/iso-grk2.ent" $CATALOG
! ${XMLCATALOG} --noout --add "public" \
      "ISO 8879:1986//ENTITIES Added Math Symbols: Arrow Relations//EN" \
      "file://$isodir/iso-amsa.ent" $CATALOG
! ${XMLCATALOG} --noout --add "public" \
      "ISO 8879:1986//ENTITIES Added Math Symbols: Ordinary//EN" \
      "file://$isodir/iso-amso.ent" $CATALOG
! ${XMLCATALOG} --noout --add "public" \
      "ISO 8879:1986//ENTITIES Russian Cyrillic//EN" \
      "file://$isodir/iso-cyr1.ent" $CATALOG
! ${XMLCATALOG} --noout --add "public" \
      "ISO 8879:1986//ENTITIES General Technical//EN" \
      "file://$isodir/iso-tech.ent" $CATALOG
! ${XMLCATALOG} --noout --add "public" \
      "ISO 8879:1986//ENTITIES Added Math Symbols: Delimiters//EN" \
      "file://$isodir/iso-amsc.ent" $CATALOG
! ${XMLCATALOG} --noout --add "public" \
      "ISO 8879:1986//ENTITIES Added Latin 1//EN" \
      "file://$isodir/iso-lat1.ent" $CATALOG
! ${XMLCATALOG} --noout --add "public" \
      "ISO 8879:1986//ENTITIES Added Math Symbols: Binary Operators//EN" \
      "file://$isodir/iso-amsb.ent" $CATALOG
! ${XMLCATALOG} --noout --add "public" \
      "ISO 8879:1986//ENTITIES Added Latin 2//EN" \
      "file://$isodir/iso-lat2.ent" $CATALOG
! ${XMLCATALOG} --noout --add "public" \
      "ISO 8879:1986//ENTITIES Added Math Symbols: Relations//EN" \
      "file://$isodir/iso-amsr.ent" $CATALOG
! ${XMLCATALOG} --noout --add "public" \
      "ISO 8879:1986//ENTITIES Non-Russian Cyrillic//EN" \
      "file://$isodir/iso-cyr2.ent" $CATALOG
  
! ${XMLCATALOG} --noout --add "delegatePublic" \
      "ISO 8879:1986" \
      "file://$CATALOG" $ROOTCATALOG
  #
