--- ./include/geos/platform.h.ORIGIN	2018-02-06 09:07:54 +0100
+++ ./include/geos/platform.h	2018-02-06 08:51:04 +0100
@@ -4,9 +4,11 @@
 
 /* Set to 1 if you have `int64_t' type */
 /* #undef HAVE_INT64_T_64 */
+#define HAVE_INT64_T_64 1
 
 /* Set to 1 if `long int' is 64 bits */
 /* #undef HAVE_LONG_INT_64 */
+#define HAVE_LONG_INT_64 1
 
 /* Set to 1 if `long long int' is 64 bits */
 #define HAVE_LONG_LONG_INT_64 1
