--- a/32bit/Makefile.ORIGIN	2019-06-21 11:09:48 +0200
+++ b/32bit/Makefile	2019-06-21 11:13:06 +0200
@@ -1158,8 +1158,8 @@
 # Set CFLAGS, LDFLAGS and other *FLAGS variables. These might be
 # tweaked by config.* below as well as the command-line, both of
 # which'll override these defaults.
-CFLAGS = -g -O2 -Wall
-LDFLAGS =
+CFLAGS = -g -O2 -Wall -D_LARGE_FILES
+LDFLAGS = -L/opt/freeware/lib -Wl,-bmaxdata:0x80000000
 BASIC_CFLAGS = -I.
 BASIC_LDFLAGS =
 
