--- Makefile.orig	2017-04-21 04:32:00 -0500
+++ Makefile	2017-04-21 04:35:04 -0500
@@ -396,6 +396,7 @@
 # CFLAGS and LDFLAGS are for the users to override from the command line.
 
 CFLAGS = -g -O2 -Wall
+CFLAGS = -std=gnu99
 DEVELOPER_CFLAGS = -Werror \
 	-Wdeclaration-after-statement \
 	-Wno-format-zero-length \
@@ -427,7 +428,7 @@
 #   htmldir
 # This can help installing the suite in a relocatable way.
 
-prefix = $(HOME)
+prefix = /opt/freeware
 bindir_relative = bin
 bindir = $(prefix)/$(bindir_relative)
 mandir = $(prefix)/share/man
@@ -451,7 +452,7 @@
 
 export prefix bindir sharedir sysconfdir gitwebdir localedir
 
-CC = cc
+CC = gcc
 AR = ar
 RM = rm -f
 DIFF = diff
