diff --git a/builtins/wait.def b/builtins/wait.def
--- a/builtins/wait.def
+++ b/builtins/wait.def
@@ -213,11 +213,11 @@ wait_builtin (list)
 	}
 
       status = wait_for_any_job (wflags, &pstat);
-      if (status < 0)
-	status = 127;
-
       if (vname && status >= 0)
 	bind_var_to_int (vname, pstat.pid);
+
+      if (status < 0)
+	status = 127;
       if (list)
 	unset_waitlist ();
       WAIT_RETURN (status);
diff --git a/patchlevel.h b/patchlevel.h
--- a/patchlevel.h
+++ b/patchlevel.h
@@ -25,6 +25,6 @@
    regexp `^#define[ 	]*PATCHLEVEL', since that's what support/mkversion.sh
    looks for to find the patch level (for the sccs version string). */
 
-#define PATCHLEVEL 1
+#define PATCHLEVEL 2
 
 #endif /* _PATCHLEVEL_H_ */
