diff -uNr fontconfig-2.13.92.orig/conf.d/05-reset-dirs-sample.conf fontconfig-2.13.92/conf.d/05-reset-dirs-sample.conf
--- fontconfig-2.13.92.orig/conf.d/05-reset-dirs-sample.conf	2019-05-08 10:22:25.000000000 +0200
+++ fontconfig-2.13.92/conf.d/05-reset-dirs-sample.conf	2020-03-28 15:53:41.792757065 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <description>Re-define fonts dirs sample</description>
 
diff -uNr fontconfig-2.13.92.orig/conf.d/09-autohint-if-no-hinting.conf fontconfig-2.13.92/conf.d/09-autohint-if-no-hinting.conf
--- fontconfig-2.13.92.orig/conf.d/09-autohint-if-no-hinting.conf	2019-07-30 13:03:27.000000000 +0200
+++ fontconfig-2.13.92/conf.d/09-autohint-if-no-hinting.conf	2020-03-28 15:53:41.764757635 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/conf.d/10-autohint.conf fontconfig-2.13.92/conf.d/10-autohint.conf
--- fontconfig-2.13.92.orig/conf.d/10-autohint.conf	2018-06-05 12:36:38.000000000 +0200
+++ fontconfig-2.13.92/conf.d/10-autohint.conf	2020-03-28 15:53:41.829756312 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/conf.d/10-hinting-full.conf fontconfig-2.13.92/conf.d/10-hinting-full.conf
--- fontconfig-2.13.92.orig/conf.d/10-hinting-full.conf	2018-06-05 12:36:38.000000000 +0200
+++ fontconfig-2.13.92/conf.d/10-hinting-full.conf	2020-03-28 15:53:41.714758653 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/conf.d/10-hinting-medium.conf fontconfig-2.13.92/conf.d/10-hinting-medium.conf
--- fontconfig-2.13.92.orig/conf.d/10-hinting-medium.conf	2018-06-05 12:36:38.000000000 +0200
+++ fontconfig-2.13.92/conf.d/10-hinting-medium.conf	2020-03-28 15:53:41.809756719 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/conf.d/10-hinting-none.conf fontconfig-2.13.92/conf.d/10-hinting-none.conf
--- fontconfig-2.13.92.orig/conf.d/10-hinting-none.conf	2018-06-05 12:36:38.000000000 +0200
+++ fontconfig-2.13.92/conf.d/10-hinting-none.conf	2020-03-28 15:53:41.796756984 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/conf.d/10-hinting-slight.conf fontconfig-2.13.92/conf.d/10-hinting-slight.conf
--- fontconfig-2.13.92.orig/conf.d/10-hinting-slight.conf	2018-06-05 12:36:38.000000000 +0200
+++ fontconfig-2.13.92/conf.d/10-hinting-slight.conf	2020-03-28 15:53:41.760757717 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/conf.d/10-no-sub-pixel.conf fontconfig-2.13.92/conf.d/10-no-sub-pixel.conf
--- fontconfig-2.13.92.orig/conf.d/10-no-sub-pixel.conf	2018-06-05 12:36:38.000000000 +0200
+++ fontconfig-2.13.92/conf.d/10-no-sub-pixel.conf	2020-03-28 15:53:41.743758063 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/conf.d/10-scale-bitmap-fonts.conf fontconfig-2.13.92/conf.d/10-scale-bitmap-fonts.conf
--- fontconfig-2.13.92.orig/conf.d/10-scale-bitmap-fonts.conf	2018-06-05 12:36:38.000000000 +0200
+++ fontconfig-2.13.92/conf.d/10-scale-bitmap-fonts.conf	2020-03-28 15:53:41.727758389 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/conf.d/10-sub-pixel-bgr.conf fontconfig-2.13.92/conf.d/10-sub-pixel-bgr.conf
--- fontconfig-2.13.92.orig/conf.d/10-sub-pixel-bgr.conf	2018-06-05 12:36:38.000000000 +0200
+++ fontconfig-2.13.92/conf.d/10-sub-pixel-bgr.conf	2020-03-28 15:53:41.841756068 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/conf.d/10-sub-pixel-rgb.conf fontconfig-2.13.92/conf.d/10-sub-pixel-rgb.conf
--- fontconfig-2.13.92.orig/conf.d/10-sub-pixel-rgb.conf	2018-06-05 12:36:38.000000000 +0200
+++ fontconfig-2.13.92/conf.d/10-sub-pixel-rgb.conf	2020-03-28 15:53:41.723758470 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/conf.d/10-sub-pixel-vbgr.conf fontconfig-2.13.92/conf.d/10-sub-pixel-vbgr.conf
--- fontconfig-2.13.92.orig/conf.d/10-sub-pixel-vbgr.conf	2018-06-05 12:36:38.000000000 +0200
+++ fontconfig-2.13.92/conf.d/10-sub-pixel-vbgr.conf	2020-03-28 15:53:41.772757472 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/conf.d/10-sub-pixel-vrgb.conf fontconfig-2.13.92/conf.d/10-sub-pixel-vrgb.conf
--- fontconfig-2.13.92.orig/conf.d/10-sub-pixel-vrgb.conf	2018-06-05 12:36:38.000000000 +0200
+++ fontconfig-2.13.92/conf.d/10-sub-pixel-vrgb.conf	2020-03-28 15:53:41.706758816 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/conf.d/10-unhinted.conf fontconfig-2.13.92/conf.d/10-unhinted.conf
--- fontconfig-2.13.92.orig/conf.d/10-unhinted.conf	2018-06-05 12:36:38.000000000 +0200
+++ fontconfig-2.13.92/conf.d/10-unhinted.conf	2020-03-28 15:53:41.739758145 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/conf.d/11-lcdfilter-default.conf fontconfig-2.13.92/conf.d/11-lcdfilter-default.conf
--- fontconfig-2.13.92.orig/conf.d/11-lcdfilter-default.conf	2018-06-05 12:36:38.000000000 +0200
+++ fontconfig-2.13.92/conf.d/11-lcdfilter-default.conf	2020-03-28 15:53:41.731758307 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/conf.d/11-lcdfilter-legacy.conf fontconfig-2.13.92/conf.d/11-lcdfilter-legacy.conf
--- fontconfig-2.13.92.orig/conf.d/11-lcdfilter-legacy.conf	2018-06-05 12:36:38.000000000 +0200
+++ fontconfig-2.13.92/conf.d/11-lcdfilter-legacy.conf	2020-03-28 15:53:41.837756149 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/conf.d/11-lcdfilter-light.conf fontconfig-2.13.92/conf.d/11-lcdfilter-light.conf
--- fontconfig-2.13.92.orig/conf.d/11-lcdfilter-light.conf	2018-06-05 12:36:38.000000000 +0200
+++ fontconfig-2.13.92/conf.d/11-lcdfilter-light.conf	2020-03-28 15:53:41.768757554 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/conf.d/20-unhint-small-vera.conf fontconfig-2.13.92/conf.d/20-unhint-small-vera.conf
--- fontconfig-2.13.92.orig/conf.d/20-unhint-small-vera.conf	2018-06-05 12:36:38.000000000 +0200
+++ fontconfig-2.13.92/conf.d/20-unhint-small-vera.conf	2020-03-28 15:53:41.784757228 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/conf.d/25-unhint-nonlatin.conf fontconfig-2.13.92/conf.d/25-unhint-nonlatin.conf
--- fontconfig-2.13.92.orig/conf.d/25-unhint-nonlatin.conf	2018-06-05 12:36:38.000000000 +0200
+++ fontconfig-2.13.92/conf.d/25-unhint-nonlatin.conf	2020-03-28 15:53:41.817756556 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/conf.d/30-metric-aliases.conf fontconfig-2.13.92/conf.d/30-metric-aliases.conf
--- fontconfig-2.13.92.orig/conf.d/30-metric-aliases.conf	2018-06-05 12:36:38.000000000 +0200
+++ fontconfig-2.13.92/conf.d/30-metric-aliases.conf	2020-03-28 15:53:41.718758572 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/conf.d/40-nonlatin.conf fontconfig-2.13.92/conf.d/40-nonlatin.conf
--- fontconfig-2.13.92.orig/conf.d/40-nonlatin.conf	2019-05-08 10:22:25.000000000 +0200
+++ fontconfig-2.13.92/conf.d/40-nonlatin.conf	2020-03-28 15:53:41.735758226 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/conf.d/45-generic.conf fontconfig-2.13.92/conf.d/45-generic.conf
--- fontconfig-2.13.92.orig/conf.d/45-generic.conf	2018-06-05 12:36:38.000000000 +0200
+++ fontconfig-2.13.92/conf.d/45-generic.conf	2020-03-28 15:53:41.833756231 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/conf.d/45-latin.conf fontconfig-2.13.92/conf.d/45-latin.conf
--- fontconfig-2.13.92.orig/conf.d/45-latin.conf	2019-05-08 10:22:25.000000000 +0200
+++ fontconfig-2.13.92/conf.d/45-latin.conf	2020-03-28 15:53:41.756757798 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/conf.d/49-sansserif.conf fontconfig-2.13.92/conf.d/49-sansserif.conf
--- fontconfig-2.13.92.orig/conf.d/49-sansserif.conf	2018-06-05 12:36:38.000000000 +0200
+++ fontconfig-2.13.92/conf.d/49-sansserif.conf	2020-03-28 15:53:41.845755987 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/conf.d/50-user.conf fontconfig-2.13.92/conf.d/50-user.conf
--- fontconfig-2.13.92.orig/conf.d/50-user.conf	2018-06-05 12:36:38.000000000 +0200
+++ fontconfig-2.13.92/conf.d/50-user.conf	2020-03-28 15:53:41.710758735 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/conf.d/51-local.conf fontconfig-2.13.92/conf.d/51-local.conf
--- fontconfig-2.13.92.orig/conf.d/51-local.conf	2018-06-05 12:36:38.000000000 +0200
+++ fontconfig-2.13.92/conf.d/51-local.conf	2020-03-28 15:53:41.805756801 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/conf.d/60-generic.conf fontconfig-2.13.92/conf.d/60-generic.conf
--- fontconfig-2.13.92.orig/conf.d/60-generic.conf	2018-06-05 12:36:38.000000000 +0200
+++ fontconfig-2.13.92/conf.d/60-generic.conf	2020-03-28 15:53:41.702758898 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/conf.d/60-latin.conf fontconfig-2.13.92/conf.d/60-latin.conf
--- fontconfig-2.13.92.orig/conf.d/60-latin.conf	2019-05-08 10:22:25.000000000 +0200
+++ fontconfig-2.13.92/conf.d/60-latin.conf	2020-03-28 15:53:41.752757880 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/conf.d/65-fonts-persian.conf fontconfig-2.13.92/conf.d/65-fonts-persian.conf
--- fontconfig-2.13.92.orig/conf.d/65-fonts-persian.conf	2018-06-05 12:36:38.000000000 +0200
+++ fontconfig-2.13.92/conf.d/65-fonts-persian.conf	2020-03-28 15:53:41.748757961 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <!--
   fonts-persian.conf
   To configure Persian fonts from The FarsiWeb Project.
diff -uNr fontconfig-2.13.92.orig/conf.d/65-khmer.conf fontconfig-2.13.92/conf.d/65-khmer.conf
--- fontconfig-2.13.92.orig/conf.d/65-khmer.conf	2018-06-05 12:36:38.000000000 +0200
+++ fontconfig-2.13.92/conf.d/65-khmer.conf	2020-03-28 15:53:41.813756638 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/conf.d/65-nonlatin.conf fontconfig-2.13.92/conf.d/65-nonlatin.conf
--- fontconfig-2.13.92.orig/conf.d/65-nonlatin.conf	2019-05-08 10:22:25.000000000 +0200
+++ fontconfig-2.13.92/conf.d/65-nonlatin.conf	2020-03-28 15:53:41.825756394 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/conf.d/69-unifont.conf fontconfig-2.13.92/conf.d/69-unifont.conf
--- fontconfig-2.13.92.orig/conf.d/69-unifont.conf	2018-06-05 12:36:38.000000000 +0200
+++ fontconfig-2.13.92/conf.d/69-unifont.conf	2020-03-28 15:53:41.821756475 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/conf.d/70-no-bitmaps.conf fontconfig-2.13.92/conf.d/70-no-bitmaps.conf
--- fontconfig-2.13.92.orig/conf.d/70-no-bitmaps.conf	2018-06-05 12:36:38.000000000 +0200
+++ fontconfig-2.13.92/conf.d/70-no-bitmaps.conf	2020-03-28 15:53:41.788757147 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/conf.d/70-yes-bitmaps.conf fontconfig-2.13.92/conf.d/70-yes-bitmaps.conf
--- fontconfig-2.13.92.orig/conf.d/70-yes-bitmaps.conf	2018-06-05 12:36:38.000000000 +0200
+++ fontconfig-2.13.92/conf.d/70-yes-bitmaps.conf	2020-03-28 15:53:41.780757310 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/conf.d/80-delicious.conf fontconfig-2.13.92/conf.d/80-delicious.conf
--- fontconfig-2.13.92.orig/conf.d/80-delicious.conf	2018-06-05 12:36:38.000000000 +0200
+++ fontconfig-2.13.92/conf.d/80-delicious.conf	2020-03-28 15:53:41.800756903 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/conf.d/90-synthetic.conf fontconfig-2.13.92/conf.d/90-synthetic.conf
--- fontconfig-2.13.92.orig/conf.d/90-synthetic.conf	2018-07-19 05:14:39.000000000 +0200
+++ fontconfig-2.13.92/conf.d/90-synthetic.conf	2020-03-28 15:53:41.776757391 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
     <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
diff -uNr fontconfig-2.13.92.orig/doc/fontconfig-user.html fontconfig-2.13.92/doc/fontconfig-user.html
--- fontconfig-2.13.92.orig/doc/fontconfig-user.html	2019-08-09 13:12:38.000000000 +0200
+++ fontconfig-2.13.92/doc/fontconfig-user.html	2020-03-28 15:53:41.694759061 +0100
@@ -422,7 +422,7 @@
 ><PRE
 CLASS="PROGRAMLISTING"
 >	&#60;?xml version="1.0"?&#62;
-	&#60;!DOCTYPE fontconfig SYSTEM "fonts.dtd"&#62;
+	&#60;!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd"&#62;
 	&#60;fontconfig&#62;
 	...
 	&#60;/fontconfig&#62;
@@ -1240,7 +1240,7 @@
 ><PRE
 CLASS="PROGRAMLISTING"
 >&#60;?xml version="1.0"?&#62;
-&#60;!DOCTYPE fontconfig SYSTEM "fonts.dtd"&#62;
+&#60;!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd"&#62;
 &#60;!-- /etc/fonts/fonts.conf file to configure system font access --&#62;
 &#60;fontconfig&#62;
 &#60;!-- 
@@ -1367,7 +1367,7 @@
 ><PRE
 CLASS="PROGRAMLISTING"
 >&#60;?xml version="1.0"?&#62;
-&#60;!DOCTYPE fontconfig SYSTEM "fonts.dtd"&#62;
+&#60;!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd"&#62;
 &#60;!-- $XDG_CONFIG_HOME/fontconfig/fonts.conf for per-user font configuration --&#62;
 &#60;fontconfig&#62;
 
diff -uNr fontconfig-2.13.92.orig/doc/fontconfig-user.sgml fontconfig-2.13.92/doc/fontconfig-user.sgml
--- fontconfig-2.13.92.orig/doc/fontconfig-user.sgml	2018-11-15 13:20:50.000000000 +0100
+++ fontconfig-2.13.92/doc/fontconfig-user.sgml	2020-03-28 15:53:41.680759345 +0100
@@ -300,7 +300,7 @@
 following structure:
     <programlisting>
 	&lt;?xml version="1.0"?&gt;
-	&lt;!DOCTYPE fontconfig SYSTEM "fonts.dtd"&gt;
+	&lt;!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd"&gt;
 	&lt;fontconfig&gt;
 	...
 	&lt;/fontconfig&gt;
@@ -573,7 +573,7 @@
     </para>
     <programlisting>
 &lt;?xml version="1.0"?&gt;
-&lt;!DOCTYPE fontconfig SYSTEM "fonts.dtd"&gt;
+&lt;!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd"&gt;
 &lt;!-- &confdir;/fonts.conf file to configure system font access --&gt;
 &lt;fontconfig&gt;
 &lt;!-- 
@@ -684,7 +684,7 @@
     </para>
     <programlisting>
 &lt;?xml version="1.0"?&gt;
-&lt;!DOCTYPE fontconfig SYSTEM "fonts.dtd"&gt;
+&lt;!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd"&gt;
 &lt;!-- $XDG_CONFIG_HOME/fontconfig/fonts.conf for per-user font configuration --&gt;
 &lt;fontconfig&gt;
 
diff -uNr fontconfig-2.13.92.orig/doc/fontconfig-user.txt fontconfig-2.13.92/doc/fontconfig-user.txt
--- fontconfig-2.13.92.orig/doc/fontconfig-user.txt	2019-08-09 13:12:20.000000000 +0200
+++ fontconfig-2.13.92/doc/fontconfig-user.txt	2020-03-28 15:53:41.689759162 +0100
@@ -261,7 +261,7 @@
    following structure:
 
            <?xml version="1.0"?>
-           <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+           <!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
            <fontconfig>
            ...
            </fontconfig>
@@ -573,7 +573,7 @@
    This is an example of a system-wide configuration file
 
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <!-- /etc/fonts/fonts.conf file to configure system font access -->
 <fontconfig>
 <!--
@@ -683,7 +683,7 @@
    $XDG_CONFIG_HOME/fontconfig/fonts.conf
 
  <?xml version="1.0"?>
- <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+ <!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
  <!-- $XDG_CONFIG_HOME/fontconfig/fonts.conf for per-user font configuration -->
  <fontconfig>
 
diff -uNr fontconfig-2.13.92.orig/doc/fonts-conf.5 fontconfig-2.13.92/doc/fonts-conf.5
--- fontconfig-2.13.92.orig/doc/fonts-conf.5	2019-08-09 13:12:20.000000000 +0200
+++ fontconfig-2.13.92/doc/fonts-conf.5	2020-03-28 15:53:41.685759244 +0100
@@ -264,7 +264,7 @@
 .sp
 .nf
 	<?xml version="1.0"?>
-	<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+	<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 	<fontconfig>
 \&...
 	</fontconfig>
@@ -554,7 +554,7 @@
 .sp
 .nf
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <!-- /etc/fonts/fonts.conf file to configure system font access -->
 <fontconfig>
 <!-- 
@@ -666,7 +666,7 @@
 .sp
 .nf
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <!-- $XDG_CONFIG_HOME/fontconfig/fonts.conf for per-user font configuration -->
 <fontconfig>
 
diff -uNr fontconfig-2.13.92.orig/fonts.conf.in fontconfig-2.13.92/fonts.conf.in
--- fontconfig-2.13.92.orig/fonts.conf.in	2019-05-08 10:22:25.000000000 +0200
+++ fontconfig-2.13.92/fonts.conf.in	2020-03-28 15:53:41.698758979 +0100
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <!-- /etc/fonts/fonts.conf file to configure system font access -->
 <fontconfig>
 	<its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
