--- ./include/my_inttypes.h.ORIGIN	2019-06-04 09:04:49 -0500
+++ ./include/my_inttypes.h	2019-06-04 09:06:20 -0500
@@ -66,10 +66,17 @@
   [unsigned] long long unconditionally in many places,
   for example in constants with [U]LL suffix.
 */
+#ifndef _AIX
 typedef unsigned long long int ulonglong; /* ulong or unsigned long long */
 typedef long long int longlong;
 typedef longlong int64;
 typedef ulonglong uint64;
+#else
+typedef unsigned long long int ulonglong; /* ulong or unsigned long long */
+typedef long long int longlong;
+typedef long int64;
+typedef unsigned long uint64;
+#endif
 
 #if defined(_WIN32)
 typedef unsigned __int64 my_ulonglong;
