--- ./mysql-test/CMakeLists.txt.ORIGIN    2019-11-22 17:08:07 +0100
+++ ./mysql-test/CMakeLists.txt 2019-11-25 13:38:54 +0100
@@ -62,7 +62,7 @@
 IF(NOT ${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_BINARY_DIR})
   # Enable running mtr from build directory
   FIND_PROGRAM(PERL_EXECUTABLE perl
-    /usr/bin /usr/local/bin
+    /opt/freeware/bin
     NO_CMAKE_ENVIRONMENT_PATH
     NO_SYSTEM_ENVIRONMENT_PATH)
   # If not found in hard-coded list above, look in PATH as well.
--- ./scripts/CMakeLists.txt.ORIGIN	2019-08-08 09:42:09 -0500
+++ ./scripts/CMakeLists.txt	2019-08-08 09:45:40 -0500
@@ -275,6 +275,8 @@
 
 IF(CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
   SET (PERL_PATH "/usr/local/bin/perl")
+ELSEIF(CMAKE_SYSTEM_NAME MATCHES "AIX")
+  SET (PERL_PATH "/opt/freeware/bin/perl")
 ELSE()
   SET (PERL_PATH "/usr/bin/perl")
 ENDIF()
