Index: libxslt-1.1.26/config.guess
===================================================================
--- libxslt-1.1.26.orig/config.guess	2008-01-30 21:27:02.000000000 +0100
+++ libxslt-1.1.26/config.guess	2012-07-03 12:55:59.000000000 +0200
@@ -532,7 +532,7 @@
 		echo rs6000-ibm-aix3.2
 	fi
 	exit ;;
-    *:AIX:*:[45])
+    *:AIX:*:[4567])
 	IBM_CPU_ID=`/usr/sbin/lsdev -C -c processor -S available | sed 1q | awk '{ print $1 }'`
 	if /usr/sbin/lsattr -El ${IBM_CPU_ID} | grep ' POWER' >/dev/null 2>&1; then
 		IBM_ARCH=rs6000
