Index: rrdtool-1.7.0/config.rpath
===================================================================
--- rrdtool-1.7.0/config.rpath.orig	2015-03-07 18:48
+++ rrdtool-1.7.0/config.rpath	2018-02-14 14:00
@@ -158,7 +158,7 @@
   # option of GNU ld is called -rpath, not --rpath.
   hardcode_libdir_flag_spec='${wl}-rpath ${wl}$libdir'
   case "$host_os" in
-    aix3* | aix4* | aix5*)
+    aix3* | aix4* | aix5* | aix6* | aix7*)
       # On AIX/PPC, the GNU linker is very broken
       if test "$host_cpu" != ia64; then
         ld_shlibs=no
@@ -254,7 +254,7 @@
         hardcode_direct=unsupported
       fi
       ;;
-    aix4* | aix5*)
+    aix4* | aix5* | aix6* | aix7*)
       if test "$host_cpu" = ia64; then
         # On IA64, the linker does run time linking by default, so we don't
         # have to do anything special.
@@ -264,7 +264,7 @@
         # Test if we are trying to use run time linking or normal
         # AIX style linking. If -brtl is somewhere in LDFLAGS, we
         # need to do runtime linking.
-        case $host_os in aix4.[23]|aix4.[23].*|aix5*)
+        case $host_os in aix4.[23]|aix4.[23].*|aix5*|aix6*|aix7*)
           for ld_flag in $LDFLAGS; do
             if (test $ld_flag = "-brtl" || test $ld_flag = "-Wl,-brtl"); then
               aix_use_runtimelinking=yes
@@ -505,7 +505,7 @@
   aix3*)
     library_names_spec='$libname.a'
     ;;
-  aix4* | aix5*)
+  aix4* | aix5* | aix6* | aix7*)
     library_names_spec='$libname$shrext'
     ;;
   amigaos*)
