From fc0027df8b06bf9cedadc50f6add67c079f8460e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Cl=C3=A9ment=20Chigot?= <clement.chigot@atos.net>
Date: Fri, 5 Jun 2020 11:36:40 -0500
Subject: [PATCH] libgo/go/runtime: add noinline for callers

---
 libgo/go/runtime/traceback_gccgo.go | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libgo/go/runtime/traceback_gccgo.go b/libgo/go/runtime/traceback_gccgo.go
index 1ba91af76ae..f760cc95cfe 100644
--- a/libgo/go/runtime/traceback_gccgo.go
+++ b/libgo/go/runtime/traceback_gccgo.go
@@ -56,6 +56,7 @@ func c_callers(skip int32, locbuf *location, max int32, keepThunks bool) int32
 
 // callers returns a stack trace of the current goroutine.
 // The gc version of callers takes []uintptr, but we take []location.
+//go:noinline
 func callers(skip int, locbuf []location) int {
 	n := c_callers(int32(skip)+1, &locbuf[0], int32(len(locbuf)), false)
 	return int(n)
-- 
2.25.0

