--- third_party/waf/waflib/Tools/gcc.py.gcc	2019-12-06 11:10:31 +0100
+++ third_party/waf/waflib/Tools/gcc.py	2020-04-10 17:31:00 +0200
@@ -44,9 +44,9 @@
 	v.STLIBPATH_ST        = '-L%s'
 	v.RPATH_ST            = '-Wl,-rpath,%s'
 
-	v.SONAME_ST           = '-Wl,-h,%s'
-	v.SHLIB_MARKER        = '-Wl,-Bdynamic'
-	v.STLIB_MARKER        = '-Wl,-Bstatic'
+	v.SONAME_ST           = ''
+	v.SHLIB_MARKER        = '-shared'
+	v.STLIB_MARKER        = ''
 
 	v.cprogram_PATTERN    = '%s'
 
@@ -54,7 +54,7 @@
 	v.LINKFLAGS_cshlib    = ['-shared']
 	v.cshlib_PATTERN      = 'lib%s.so'
 
-	v.LINKFLAGS_cstlib    = ['-Wl,-Bstatic']
+	v.LINKFLAGS_cstlib    = ['']
 	v.cstlib_PATTERN      = 'lib%s.a'
 
 	v.LINKFLAGS_MACBUNDLE = ['-bundle', '-undefined', 'dynamic_lookup']
@@ -69,7 +69,7 @@
 
 	v.cshlib_PATTERN      = '%s.dll'
 	v.implib_PATTERN      = '%s.dll.a'
-	v.IMPLIB_ST           = '-Wl,--out-implib,%s'
+	v.IMPLIB_ST           = '-Wl,-bI:%s'
 
 	v.CFLAGS_cshlib       = []
 
@@ -109,7 +109,7 @@
 	"""Configuration flags for executing gcc on AIX"""
 	v = conf.env
 	v.LINKFLAGS_cprogram  = ['-Wl,-brtl']
-	v.LINKFLAGS_cshlib    = ['-shared','-Wl,-brtl,-bexpfull']
+	v.LINKFLAGS_cshlib    = ['-shared','-Wl,-brtl,-bexpfull,-berok']
 	v.SHLIB_MARKER        = []
 
 @conf
--- third_party/waf/waflib/Tools/gxx.py.gcc	2019-12-06 11:10:31 +0100
+++ third_party/waf/waflib/Tools/gxx.py	2020-04-10 17:30:52 +0200
@@ -44,9 +44,9 @@
 	v.STLIBPATH_ST        = '-L%s'
 	v.RPATH_ST            = '-Wl,-rpath,%s'
 
-	v.SONAME_ST           = '-Wl,-h,%s'
-	v.SHLIB_MARKER        = '-Wl,-Bdynamic'
-	v.STLIB_MARKER        = '-Wl,-Bstatic'
+	v.SONAME_ST           = ''
+	v.SHLIB_MARKER        = '-shared'
+	v.STLIB_MARKER        = ''
 
 	v.cxxprogram_PATTERN  = '%s'
 
@@ -54,7 +54,7 @@
 	v.LINKFLAGS_cxxshlib  = ['-shared']
 	v.cxxshlib_PATTERN    = 'lib%s.so'
 
-	v.LINKFLAGS_cxxstlib  = ['-Wl,-Bstatic']
+	v.LINKFLAGS_cxxstlib  = ['']
 	v.cxxstlib_PATTERN    = 'lib%s.a'
 
 	v.LINKFLAGS_MACBUNDLE = ['-bundle', '-undefined', 'dynamic_lookup']
@@ -69,7 +69,7 @@
 
 	v.cxxshlib_PATTERN    = '%s.dll'
 	v.implib_PATTERN      = '%s.dll.a'
-	v.IMPLIB_ST           = '-Wl,--out-implib,%s'
+	v.IMPLIB_ST           = '-Wl,-bI:%s'
 
 	v.CXXFLAGS_cxxshlib   = []
 
@@ -110,7 +110,7 @@
 	v = conf.env
 	v.LINKFLAGS_cxxprogram= ['-Wl,-brtl']
 
-	v.LINKFLAGS_cxxshlib  = ['-shared', '-Wl,-brtl,-bexpfull']
+	v.LINKFLAGS_cxxshlib  = ['-shared', '-Wl,-brtl,-bexpfull,-berok']
 	v.SHLIB_MARKER        = []
 
 @conf
