--- ./lib/tdb/common/io.c_orig	2019-01-22 17:23:32 +0000
+++ ./lib/tdb/common/io.c	2019-01-23 18:40:10 +0000
@@ -96,7 +96,7 @@
 	return ret;
 }
 
-#ifdef HAVE_POSIX_FALLOCATE
+#if defined(HAVE_POSIX_FALLOCATE) && !defined(_AIX) 
 static int tdb_posix_fallocate(struct tdb_context *tdb, off_t offset,
 			       off_t len)
 {
@@ -413,7 +413,7 @@
 		return -1;
 	}
 
-#ifdef HAVE_POSIX_FALLOCATE
+#if defined(HAVE_POSIX_FALLOCATE) && !defined(_AIX)
 	ret = tdb_posix_fallocate(tdb, size, addition);
 	if (ret == 0) {
 		return 0;
