--- ./libs/math/test/Jamfile.v2.ORIGIN	2020-07-01 10:14:20 -0500
+++ ./libs/math/test/Jamfile.v2	2020-07-01 14:12:56 -0500
@@ -879,6 +879,7 @@
       ../build//boost_math_c99f
       ../../test/build//boost_unit_test_framework
       : : : #requirements
+      <target-os>aix:<linkflags>-Wl,-berok
       :
       test_tr1_c
     ]
@@ -890,6 +891,7 @@
       : : :
       <define>TEST_LD=1
       [ check-target-builds ../config//has_long_double_support "long double support" : : <build>no ]
+      <target-os>aix:<linkflags>-Wl,-berok
       :
       test_tr1_c_long_double
     ]
