--- ./libs/thread/test/Jamfile.v2.ORIGIN	2020-06-30 18:08:46 -0500
+++ ./libs/thread/test/Jamfile.v2	2020-06-30 18:14:18 -0500
@@ -121,7 +121,7 @@
 }
 
 
-rule thread-test ( sources )
+rule thread-test ( sources : more * )
 {
     sources = $(sources) winrt_init.cpp ;
     return
@@ -130,7 +130,7 @@
     ]
     [ run $(sources) ../src/tss_null.cpp ../build//boost_thread/<link>static
         : : :
-      <library>/boost/test//boost_unit_test_framework/<link>static
+      <library>/boost/test//boost_unit_test_framework/<link>static $(more)
     : $(sources[1]:B)_lib
     ]
     ;
@@ -296,7 +296,7 @@
           [ thread-test test_thread_return_local.cpp ]
           [ thread-test test_thread_move_return.cpp ]
           [ thread-test test_thread_launching.cpp ]
-          [ thread-test test_thread_mf.cpp ]
+          [ thread-test test_thread_mf.cpp : <target-os>aix:<linkflags>-Wl,-berok ]
           [ thread-test test_thread_exit.cpp ]
           [ thread-test test_move_function.cpp ]
           [ compile-fail no_implicit_move_from_lvalue_thread.cpp ]
@@ -344,7 +344,7 @@
     #explicit tickets ;
     test-suite tickets
     :
-          [ thread-test test_2309.cpp ]
+          [ thread-test test_2309.cpp : <target-os>aix:<linkflags>-Wl,-berok ]
           [ thread-run test_2501.cpp ]
           [ thread-test test_2741.cpp ]
           [ thread-run test_3628.cpp ]
@@ -366,7 +366,7 @@
           [ thread-run test_9856.cpp ]
           [ thread-compile test_10963.cpp : : test_10963_c ]
           [ thread-run test_10964.cpp ]
-          [ thread-test test_11053.cpp ]
+          [ thread-test test_11053.cpp : <target-os>aix:<linkflags>-Wl,-berok ]
           [ thread-run test_11266.cpp ]
     ;
 
