--- ./libs/test/test/Jamfile.v2.ORIGIN	2020-06-30 17:45:42 -0500
+++ ./libs/test/test/Jamfile.v2	2020-07-01 09:42:52 -0500
@@ -112,7 +112,7 @@
   [ boost.test-self-test run : utils-ts : algorithm-test ]
   [ boost.test-self-test run : utils-ts : basic_cstring-test ]
   [ boost.test-self-test run : utils-ts : class_properties-test ]
-  [ boost.test-self-test run : utils-ts : foreach-test ]
+  [ boost.test-self-test run : utils-ts : foreach-test : : : : : <target-os>aix:<linkflags>-Wl,-berok : ]
   [ boost.test-self-test run : utils-ts : named_params-test ]
   [ boost.test-self-test run : utils-ts : runtime-param-test : : : : : : [ requires cxx11_auto_declarations cxx11_function_template_default_args cxx11_lambdas cxx11_hdr_initializer_list cxx11_template_aliases ] ]
   [ boost.test-self-test run : utils-ts : string_cast-test ]
