--- ./libs/math/test/test_polynomial.cpp.ORIGIN	2020-05-15 14:00:36 -0500
+++ ./libs/math/test/test_polynomial.cpp	2020-05-18 18:44:01 -0500
@@ -265,7 +265,7 @@
 
 // Sanity checks to make sure I didn't break it.
 #ifdef TEST1
-typedef boost::mpl::list<char, short, int, long> integral_test_types;
+typedef boost::mpl::list<signed char, short, int, long> integral_test_types;
 typedef boost::mpl::list<int, long> large_integral_test_types;
 typedef boost::mpl::list<> mp_integral_test_types;
 #elif defined(TEST2)
