--- ./libs/math/test/Jamfile.v2.orig	2020-06-17 13:31:57 -0500
+++ ./libs/math/test/Jamfile.v2	2020-06-17 16:51:05 -0500
@@ -56,6 +56,7 @@
       <toolset>clang-6.0.0:<pch>off  # added to see effect.
       <toolset>gcc,<target-os>windows:<pch>off
       <toolset>borland:<runtime-link>static
+      <target-os>aix:<linkflags>"-latomic"
       # <toolset>msvc:<cxxflags>/wd4506 has no effect?
       # suppress xstring(237) : warning C4506: no definition for inline function
       <include>../../..
