Index: freetds-0.82/configure
===================================================================
--- freetds-0.82.orig/configure	2008-05-08 00:59:26.000000000 +0200
+++ freetds-0.82/configure	2011-06-22 15:42:19.000000000 +0200
@@ -22578,8 +22578,10 @@
 #define HAVE_GNUTLS 1
 _ACEOF
 
-	CPPFLAGS="$CPPFLAGS `libgnutls-config --cflags`"
-	NETWORK_LIBS="$NETWORK_LIBS `libgnutls-config --libs`"
+	## Obsolete call to libgnutls-config CPPFLAGS="$CPPFLAGS `libgnutls-config --cflags`"
+	## Obsolete call to libgnutls-config NETWORK_LIBS="$NETWORK_LIBS `libgnutls-config --libs`"
+	CPPFLAGS="$CPPFLAGS `pkg-config --cflags gnutls`"
+	NETWORK_LIBS="$NETWORK_LIBS `pkg-config --libs gnutls`"
 else
 	echo "$as_me:$LINENO: checking if openssl is wanted" >&5
 echo $ECHO_N "checking if openssl is wanted... $ECHO_C" >&6
Index: freetds-0.82/configure.ac
===================================================================
--- freetds-0.82.orig/configure.ac	2008-05-08 00:57:52.000000000 +0200
+++ freetds-0.82/configure.ac	2011-06-22 15:42:10.000000000 +0200
@@ -623,8 +623,10 @@
 AS_HELP_STRING([--with-gnutls], [build with TLS support]))
 if test "$with_gnutls" = "yes"; then
 	AC_DEFINE(HAVE_GNUTLS, 1, [Define to 1 if you have GNU tls.])
-	CPPFLAGS="$CPPFLAGS `libgnutls-config --cflags`"
-	NETWORK_LIBS="$NETWORK_LIBS `libgnutls-config --libs`"
+	## Obsolete call to libgnutls-config CPPFLAGS="$CPPFLAGS `libgnutls-config --cflags`"
+	## Obsolete call to libgnutls-config NETWORK_LIBS="$NETWORK_LIBS `libgnutls-config --libs`"
+	CPPFLAGS="$CPPFLAGS `pkg-config --cflags gnutls`"
+	NETWORK_LIBS="$NETWORK_LIBS `pkg-config --libs gnutls`"
 else
 	CHECK_OPENSSL
 fi
