--- src/jansson_config.h.in.orig	2018-02-09 08:55:15 +0100
+++ src/jansson_config.h.in	2018-10-05 16:54:50 +0200
@@ -38,11 +38,11 @@
 
 /* If __atomic builtins are available they will be used to manage
    reference counts of json_t. */
-#define JSON_HAVE_ATOMIC_BUILTINS @json_have_atomic_builtins@
+#define JSON_HAVE_ATOMIC_BUILTINS 0
 
 /* If __atomic builtins are not available we try using __sync builtins
    to manage reference counts of json_t. */
-#define JSON_HAVE_SYNC_BUILTINS @json_have_sync_builtins@
+#define JSON_HAVE_SYNC_BUILTINS 0
 
 /* Maximum recursion depth for parsing JSON input.
    This limits the depth of e.g. array-within-array constructions. */
