--- ./byterun/Makefile.ORIGIN	2018-08-24 14:11:36 +0200
+++ ./byterun/Makefile	2018-08-24 14:11:50 +0200
@@ -36,11 +36,11 @@
 		  prims.o libcamlrund.a $(BYTECCLIBS)
 
 libcamlrun.a: $(OBJS)
-	ar rc libcamlrun.a $(OBJS)
+	/usr/bin/ar rc libcamlrun.a $(OBJS)
 	$(RANLIB) libcamlrun.a
 
 libcamlrund.a: $(DOBJS)
-	ar rc libcamlrund.a $(DOBJS)
+	/usr/bin/ar rc libcamlrund.a $(DOBJS)
 	$(RANLIB) libcamlrund.a
 
 libcamlrun_shared.so: $(PICOBJS)
