diff -crN ./libexif/libexif/Makefile.am /gestconf/project/GNOME_ACL/GNOME/build/sh_build_GNOME/src/./libexif/libexif/Makefile.am
*** ./libexif/libexif/Makefile.am	Mon Mar 17 21:38:46 2003
--- /gestconf/project/GNOME_ACL/GNOME/build/sh_build_GNOME/src/./libexif/libexif/Makefile.am	Mon Feb 13 15:20:35 2006
***************
*** 19,25 ****
  	exif-tag.c		\
  	exif-utils.c		\
  	i18n.h
! libexif_la_LIBADD = -lm
  
  libexifincludedir = $(includedir)/libexif
  libexifinclude_HEADERS = 	\
--- 19,25 ----
  	exif-tag.c		\
  	exif-utils.c		\
  	i18n.h
! libexif_la_LIBADD = -lm $(LTLIBINTL)
  
  libexifincludedir = $(includedir)/libexif
  libexifinclude_HEADERS = 	\
