--- a/metricbeat/magefile.go	2020-01-09 09:46:03 +0100
+++ b/metricbeat/magefile.go	2020-01-09 09:46:20 +0100
@@ -32,7 +32,7 @@
 	metricbeat "github.com/elastic/beats/metricbeat/scripts/mage"
 
 	// mage:import
-	build "github.com/elastic/beats/dev-tools/mage/target/build"
+	// build "github.com/elastic/beats/dev-tools/mage/target/build"
 	// mage:import
 	"github.com/elastic/beats/dev-tools/mage/target/common"
 	// mage:import
@@ -72,7 +72,7 @@
 	metricbeat.CustomizePackaging()
 
 	mg.Deps(update.Update, metricbeat.PrepareModulePackagingOSS)
-	mg.Deps(build.CrossBuild, build.CrossBuildGoDaemon)
+	// mg.Deps(build.CrossBuild, build.CrossBuildGoDaemon)
 	mg.SerialDeps(devtools.Package, TestPackages)
 }
 
