--- PyXML-0.8.4/extensions/expat/lib/xmlparse.c.lr	2006-07-27 15:08:44.000000000 +0200
+++ PyXML-0.8.4/extensions/expat/lib/xmlparse.c	2006-07-27 15:09:03.000000000 +0200
@@ -67,15 +67,6 @@
 /* Round up n to be a multiple of sz, where sz is a power of 2. */
 #define ROUND_UP(n, sz) (((n) + ((sz) - 1)) & ~((sz) - 1))
 
-/* Handle the case where memmove() doesn't exist. */
-#ifndef HAVE_MEMMOVE
-#ifdef HAVE_BCOPY
-#define memmove(d,s,l) bcopy((s),(d),(l))
-#else
-#error memmove does not exist on this platform, nor is a substitute available
-#endif /* HAVE_BCOPY */
-#endif /* HAVE_MEMMOVE */
-
 #include "internal.h"
 #include "xmltok.h"
 #include "xmlrole.h"
