--- ./t/01-test.t.orig	2019-10-23 09:32:02 +0200
+++ ./t/01-test.t	2019-10-23 09:14:36 +0200
@@ -420,7 +420,7 @@
 	# When we sleep 4 the above is still true, except that one of 12.2.1 machines returned 'timeout':
 	# http://www.cpantesters.org/cpan/report/6dba0d70-2d3d-11e4-8483-fe44e5e3eb0b
 	my $expected = 'timeout';
-	if ($Config{osname} =~ /^(freebsd|midnightbsd|dragonfly)$/) {
+	if ($Config{osname} =~ /^(freebsd|midnightbsd|dragonfly|aix)$/) {
 		$expected = 'eof';
 	}
 	if ($Config{osname} eq 'darwin' and $Config{osvers} lt '13') {
