--- php-7.2.16/ext/mysqlnd/mysqlnd_result.c.orig	2019-04-03 05:13:53 +0000
+++ php-7.2.16/ext/mysqlnd/mysqlnd_result.c	2019-04-03 05:14:52 +0000
@@ -29,6 +29,15 @@
 #include "mysqlnd_debug.h"
 #include "mysqlnd_ext_plugin.h"
 
+/* #if defined(_AIX51) && !defined(_AIX52) */
+#if defined(_AIX61)
+#if (__64BIT__)
+#define SIZE_MAX        UINT64_MAX
+#else
+#define SIZE_MAX        UINT32_MAX
+#endif
+#endif
+
 /* {{{ mysqlnd_result_buffered_zval::initialize_result_set_rest */
 static enum_func_status
 MYSQLND_METHOD(mysqlnd_result_buffered_zval, initialize_result_set_rest)(MYSQLND_RES_BUFFERED * const result,
