*** ./glib/gatomic.c.ORIG	Thu Feb 10 13:22:21 2011
--- ./glib/gatomic.c	Thu Feb 10 13:22:36 2011
***************
*** 27,32 ****
--- 27,33 ----
  #include <sched.h>
  #endif
  
+ #include "glib.h"
  #include "gatomic.h"
  #include "gthreadprivate.h"
  
*** ./gio/libasyncns/g-asyncns.h.ORIG	Thu Feb 10 13:59:04 2011
--- ./gio/libasyncns/g-asyncns.h	Thu Feb 10 13:59:15 2011
***************
*** 27,33 ****
--- 27,35 ----
  
  /* asyncns.c removed this for #580301, but that appears to have been wrong */
  #if HAVE_ARPA_NAMESER_COMPAT_H
+ #ifndef _AIX
  #include <arpa/nameser_compat.h>
+ #endif
  #endif
  
  #include "asyncns.h"
*** ./gio/gsocket.c.ORIG	Thu Feb 10 14:51:22 2011
--- ./gio/gsocket.c	Thu Feb 10 14:51:44 2011
***************
*** 56,61 ****
--- 56,65 ----
  #include "gcredentials.h"
  #include "glibintl.h"
  
+ #if (defined(_AIX51) || defined(_AIX52)) && !(defined(_AIX53) || defined(_AIX61))
+ #define ss_family __ss_family
+ #endif
+ 
  /**
   * SECTION:gsocket
   * @short_description: Low-level socket object
*** ./configure.ORIG	Wed Apr 11 23:08:42 2012
--- ./configure	Wed Apr 11 23:09:26 2012
***************
*** 23466,23484 ****
  fi
  done
  
- # Check for high-resolution sleep functions
- for ac_func in splice
- do :
-   ac_fn_c_check_func "$LINENO" "splice" "ac_cv_func_splice"
- if test "x$ac_cv_func_splice" = xyes; then :
-   cat >>confdefs.h <<_ACEOF
- #define HAVE_SPLICE 1
- _ACEOF
- 
- fi
- done
- 
- 
  # To avoid finding a compatibility unusable statfs, which typically
  # successfully compiles, but warns to use the newer statvfs interface:
  if test $ac_cv_header_sys_statvfs_h = yes; then :
--- 23466,23471 ----
***************
*** 26168,26173 ****
--- 26168,26177 ----
        *-*-cygwin*|*-*-darwin*)
           # skip cygwin and darwin -pthread or -pthreads test
           ;;
+       *-aix*)
+         G_THREAD_CFLAGS="-qthreaded -D_THREAD_SAFE"
+         G_THREAD_LIBS="-lpthread"
+         ;;
        *-solaris*)
          # These compiler/linker flags work with both Sun Studio and gcc
  	# Sun Studio expands -mt to -D_REENTRANT and -lthread
