Index: libffi-3.0.9/src/powerpc/aix.S
===================================================================
--- libffi-3.0.9.orig/src/powerpc/aix.S	2009-12-29 16:22:26.000000000 +0100
+++ libffi-3.0.9/src/powerpc/aix.S	2011-07-29 15:54:15.000000000 +0200
@@ -79,6 +79,8 @@
 	.set f20,20
 	.set f21,21
 
+	.extern .ffi_prep_args
+
 #define LIBFFI_ASM
 #include <fficonfig.h>
 #include <ffi.h>
@@ -125,6 +127,7 @@
 	/* Call ffi_prep_args.  */
 	mr	r4, r1
 	bl	.ffi_prep_args
+	nop
 
 	/* Now do the call.  */
 	ld	r0, 0(r29)
@@ -226,6 +229,7 @@
 	/* Call ffi_prep_args.  */
 	mr	r4, r1
 	bl	.ffi_prep_args
+	nop
 
 	/* Now do the call.  */
 	lwz	r0, 0(r29)
Index: libffi-3.0.9/src/powerpc/aix_closure.S
===================================================================
--- libffi-3.0.9.orig/src/powerpc/aix_closure.S	2009-12-29 16:22:26.000000000 +0100
+++ libffi-3.0.9/src/powerpc/aix_closure.S	2011-07-29 15:55:05.000000000 +0200
@@ -79,6 +79,8 @@
 	.set f20,20
 	.set f21,21
 
+	.extern .ffi_closure_helper_DARWIN
+
 #define LIBFFI_ASM
 #define JUMPTARGET(name) name
 #define L(x) x
