--- ./CMakeLists.txt.orig	2021-03-03 11:48:48 +0100
+++ ./CMakeLists.txt	2021-03-03 12:15:08 +0100
@@ -327,7 +327,7 @@
                                      INSTALL_RPATH_USE_LINK_PATH 0
                                      BUILD_WITH_INSTALL_RPATH 1
                                      INSTALL_RPATH "${MAODBC_INSTALL_RPATH}")
-  ELSE()
+  ELSEIF(NOT CMAKE_SYSTEM_NAME MATCHES AIX)
     SET_TARGET_PROPERTIES(${LIBRARY_NAME} PROPERTIES LINK_FLAGS "-Wl,--version-script=${CMAKE_SOURCE_DIR}/maodbc.def")
   ENDIF()
 ENDIF()
