--- ./ma_dsn.c.origin	2019-08-01 16:54:29 +0200
+++ ./ma_dsn.c	2019-08-01 16:46:32 +0200
@@ -273,7 +273,11 @@
         IntValue= 0;
         for (i= 0; i < sizeof(TlsVersionBits); ++i)
         {
+#ifdef _AIX
+          if (strstr(Value, TlsVersionName[i]) != NULL)
+#else
           if (strcasestr(Value, TlsVersionName[i]) != NULL)
+#endif
           {
             IntValue|= TlsVersionBits[i];
           }
