--- libpng-1.5.26.orig/contrib/libtests/pngvalid.c	2016-04-25 16:40:05 +0200
+++ libpng-1.5.26/contrib/libtests/pngvalid.c	2015-12-17 14:59:17 +0100
@@ -7279,13 +7279,11 @@
           * affects the limit used for checking for internal calculation errors,
           * not the actual limit imposed by pngvalid on the output errors.
           */
-         that->pm->limit += pow(
 #        if DIGITIZE
-            1.3
+         that->pm->limit += pow( 1.3 /255, data.gamma);
 #        else
-            1.0
+         that->pm->limit += pow( 1.0 /255, data.gamma);
 #        endif
-            /255, data.gamma);
       }
    }

