*** ./cups/http.c.ORIG	Wed Nov 30 11:20:31 2016
--- ./cups/http.c	Wed Nov 30 11:21:22 2016
***************
*** 3468,3474 ****
   * @since CUPS 1.7/macOS 10.9@
   */
  
! int					/* O - 0 on success, -1 on error */
  httpWriteResponse(http_t        *http,	/* I - HTTP connection */
  		  http_status_t status)	/* I - Status code */
  {
--- 3468,3474 ----
   * @since CUPS 1.7/macOS 10.9@
   */
  
! http_state_t				/* O - 0 on success, -1 on error */
  httpWriteResponse(http_t        *http,	/* I - HTTP connection */
  		  http_status_t status)	/* I - Status code */
  {
