--- ./src/test/perl/TestLib.pm~	2018-05-21 16:08:10 -0500
+++ ./src/test/perl/TestLib.pm	2018-05-29 16:50:20 -0500
@@ -16,7 +16,7 @@
 use Fcntl qw(:mode);
 use File::Basename;
 use File::Find;
-use File::Spec;
+use File::Path qw(make_path remove_tree);use File::Spec;
 use File::stat qw(stat);
 use File::Temp ();
 use IPC::Run;
@@ -92,7 +92,7 @@
 	$tmp_check = $ENV{TESTDIR} ? "$ENV{TESTDIR}/tmp_check" : "tmp_check";
 	$log_path = "$tmp_check/log";
 
-	mkdir $tmp_check;
+	make_path $tmp_check;mkdir $tmp_check;
 	mkdir $log_path;
 
 	# Open the test log file, whose name depends on the test name.
