--- ./src/backend/parser/gram.c.ORIGIN	2017-11-28 15:12:44 -0600
+++ ./src/backend/parser/gram.c	2017-11-30 15:39:49 -0600
@@ -1,3 +1,7 @@
+// That works !
+// Useful for tests: create_aggregate and aggregates with XLC v13 and -O2 in 32bit
+#pragma option_override(makeOrderedSetArgs, "OPT(LEVEL,0)")
+
 /* A Bison parser, made by GNU Bison 3.0.2.  */
 
 /* Bison implementation for Yacc-like parsers in C
@@ -232,6 +236,8 @@
 static List *extractAggrArgTypes(List *aggrargs);
 static List *makeOrderedSetArgs(List *directargs, List *orderedargs,
 								core_yyscan_t yyscanner);
+// NO effect !!
+//								core_yyscan_t yyscanner) __attribute__((noinline)) ;
 static void insertSelectOptions(SelectStmt *stmt,
 								List *sortClause, List *lockingClause,
 								Node *limitOffset, Node *limitCount,
