--- ./src/test/modules/commit_ts/commit_ts.conf.ORIGIN	2019-10-03 16:13:23 -0500
+++ ./src/test/modules/commit_ts/commit_ts.conf	2019-10-03 16:14:22 -0500
@@ -1 +1,2 @@
 track_commit_timestamp = on
+shared_memory_type = sysv
--- ./src/test/modules/snapshot_too_old/sto.conf.ORIGIN	2019-10-03 16:13:29 -0500
+++ ./src/test/modules/snapshot_too_old/sto.conf	2019-10-03 16:14:28 -0500
@@ -1,2 +1,3 @@
 autovacuum = off
 old_snapshot_threshold = 0
+shared_memory_type = sysv
--- ./src/test/modules/test_rls_hooks/rls_hooks.conf.ORIGIN	2019-10-03 16:13:36 -0500
+++ ./src/test/modules/test_rls_hooks/rls_hooks.conf	2019-10-03 16:14:32 -0500
@@ -1 +1,2 @@
 shared_preload_libraries = test_rls_hooks
+shared_memory_type = sysv
