--- ./src/Makefile.shlib.ORIGIN	2021-09-01 09:10:05 +0200
+++ ./src/Makefile.shlib	2021-09-01 10:08:28 +0200
@@ -329,7 +329,11 @@
 	rm -f $(stlib)
 	$(LINK.static) $(stlib) $^
 	$(RANLIB) $(stlib)
-	$(MKLDEXPORT) $(stlib) $(shlib) >$(exports_file)
+ifeq (,$(SHLIB_EXPORTS))
+	$(MKLDEXPORT) $(stlib) $(shlib) > $(exports_file)
+else
+	( echo '#! $(shlib)'; $(AWK) '/^[^#]/ {printf "%s\n",$$1}' $(SHLIB_EXPORTS) ) > $(exports_file)
+endif
 	$(COMPILER) -o $(shlib) $(stlib) -Wl,-bE:$(exports_file) $(LDFLAGS) $(LDFLAGS_SL) $(SHLIB_LINK)
 	rm -f $(stlib)
 	$(AR) $(AROPT) $(stlib) $(shlib)
