--- ./unix/configure.orig	2020-11-04 12:44:36 +0100
+++ ./unix/configure	2020-11-04 16:49:46 +0100
@@ -6686,6 +6686,8 @@
 
 fi
 
+		SHARED_LIB_SUFFIX='.so.${VERSION}'
+		UNSHARED_LIB_SUFFIX='.a'
 	    ;;
 	BeOS*)
 	    SHLIB_CFLAGS="-fPIC"
@@ -18815,10 +18817,14 @@
     eval libdir="$libdir"
     # default install directory for bundled packages
     PACKAGE_DIR="$libdir"
-    if test "${TCL_LIB_VERSIONS_OK}" = "ok"; then
-        TCL_LIB_FLAG="-ltcl${TCL_VERSION}"
+    if test "`uname -s`" = "AIX"; then
+        TCL_LIB_FLAG="-ltcl"
     else
-        TCL_LIB_FLAG="-ltcl`echo ${TCL_VERSION} | tr -d .`"
+        if test "${TCL_LIB_VERSIONS_OK}" = "ok"; then
+            TCL_LIB_FLAG="-ltcl${TCL_VERSION}"
+        else
+            TCL_LIB_FLAG="-ltcl`echo ${TCL_VERSION} | tr -d .`"
+        fi
     fi
     TCL_BUILD_LIB_SPEC="-L`pwd | sed -e 's/ /\\\\ /g'` ${TCL_LIB_FLAG}"
     TCL_LIB_SPEC="-L${libdir} ${TCL_LIB_FLAG}"
@@ -18858,10 +18864,14 @@
 eval "TCL_STUB_LIB_FILE=\"${TCL_STUB_LIB_FILE}\""
 eval "TCL_STUB_LIB_DIR=${libdir}"
 
-if test "${TCL_LIB_VERSIONS_OK}" = "ok"; then
-    TCL_STUB_LIB_FLAG="-ltclstub${TCL_VERSION}"
+if test "`uname -s`" = "AIX"; then
+    TCL_STUB_LIB_FLAG="-ltclstub"
 else
-    TCL_STUB_LIB_FLAG="-ltclstub`echo ${TCL_VERSION} | tr -d .`"
+    if test "${TCL_LIB_VERSIONS_OK}" = "ok"; then
+        TCL_STUB_LIB_FLAG="-ltclstub${TCL_VERSION}"
+    else
+        TCL_STUB_LIB_FLAG="-ltclstub`echo ${TCL_VERSION} | tr -d .`"
+    fi
 fi
 
 TCL_BUILD_STUB_LIB_SPEC="-L`pwd | sed -e 's/ /\\\\ /g'` ${TCL_STUB_LIB_FLAG}"
--- ./unix/configure.in.orig	2020-11-04 12:44:46 +0100
+++ ./unix/configure.in	2020-11-04 16:48:34 +0100
@@ -840,10 +840,14 @@
     eval libdir="$libdir"
     # default install directory for bundled packages
     PACKAGE_DIR="$libdir"
-    if test "${TCL_LIB_VERSIONS_OK}" = "ok"; then
-        TCL_LIB_FLAG="-ltcl${TCL_VERSION}"
+    if test "`uname -s`" = "AIX"; then
+        TCL_LIB_FLAG="-ltcl"
     else
-        TCL_LIB_FLAG="-ltcl`echo ${TCL_VERSION} | tr -d .`"
+        if test "${TCL_LIB_VERSIONS_OK}" = "ok"; then
+            TCL_LIB_FLAG="-ltcl${TCL_VERSION}"
+        else
+           TCL_LIB_FLAG="-ltcl`echo ${TCL_VERSION} | tr -d .`"
+        fi
     fi
     TCL_BUILD_LIB_SPEC="-L`pwd | sed -e 's/ /\\\\ /g'` ${TCL_LIB_FLAG}"
     TCL_LIB_SPEC="-L${libdir} ${TCL_LIB_FLAG}"
@@ -883,10 +887,13 @@
 eval "TCL_STUB_LIB_FILE=\"${TCL_STUB_LIB_FILE}\""
 eval "TCL_STUB_LIB_DIR=${libdir}"
 
-if test "${TCL_LIB_VERSIONS_OK}" = "ok"; then
-    TCL_STUB_LIB_FLAG="-ltclstub${TCL_VERSION}"
-else
-    TCL_STUB_LIB_FLAG="-ltclstub`echo ${TCL_VERSION} | tr -d .`"
+if test "`uname -s`" = "AIX"; then
+    TCL_LIB_FLAG="-ltclstub"
+    if test "${TCL_LIB_VERSIONS_OK}" = "ok"; then
+        TCL_STUB_LIB_FLAG="-ltclstub${TCL_VERSION}"
+    else
+        TCL_STUB_LIB_FLAG="-ltclstub`echo ${TCL_VERSION} | tr -d .`"
+    fi
 fi
 
 TCL_BUILD_STUB_LIB_SPEC="-L`pwd | sed -e 's/ /\\\\ /g'` ${TCL_STUB_LIB_FLAG}"
--- ./unix/tcl.m4.orig	2020-11-04 16:46:56 +0100
+++ ./unix/tcl.m4	2020-11-04 16:44:32 +0100
@@ -1178,6 +1178,8 @@
 		CC_SEARCH_FLAGS='-L${LIB_RUNTIME_DIR}'
 		LD_SEARCH_FLAGS=${CC_SEARCH_FLAGS}
 	    ])
+		SHARED_LIB_SUFFIX='.so.${VERSION}'
+		UNSHARED_LIB_SUFFIX='.a'
 	    ;;
 	BeOS*)
 	    SHLIB_CFLAGS="-fPIC"
