Index: nss-3.14.3/mozilla/security/coreconf/AIX.mk
===================================================================
--- nss-3.14.3.orig/mozilla/security/coreconf/AIX.mk	2012-05-03 11:42:50.000000000 +0200
+++ nss-3.14.3/mozilla/security/coreconf/AIX.mk	2013-02-22 15:07:29.000000000 +0100
@@ -28,20 +28,20 @@
 	PTHREADS_USER = 1
 endif
 
-DEFAULT_COMPILER = xlc_r
+DEFAULT_COMPILER = /usr/vac/bin/xlc_r
 
-CC		= xlc_r
-CCC		= xlC_r
+CC		= /usr/vac/bin/xlc_r
+CCC		= /usr/vacpp/bin/xlC_r
 
 CPU_ARCH	= rs6000
 
 RANLIB		= ranlib
 
 OS_CFLAGS	= -DAIX -DSYSV
-OS_LIBS 	+= -blibpath:/usr/lib:/lib -lc -lm
+OS_LIBS         += -blibpath:/opt/freeware/lib:/usr/lib:/lib -lc -lm
 
 DSO_LDOPTS	= -brtl -bnortllib -bM:SRE -bnoentry
-MKSHLIB 	= $(LD) $(DSO_LDOPTS) -blibpath:/usr/lib:/lib -lc -lm
+MKSHLIB         = $(LD) $(DSO_LDOPTS) -blibpath:/opt/freeware/lib:/usr/lib:/lib -lc -lm
 
 AIX_WRAP	= $(DIST)/lib/aixwrap.o
 AIX_TMP		= $(OBJDIR)/_aix_tmp.o
