--- ./libgo/testsuite/gotest.ORIGIN	2017-06-13 16:18:37 +0200
+++ ./libgo/testsuite/gotest	2017-06-13 16:26:51 +0200
@@ -543,7 +543,7 @@
         # The final sed removes the leading . added to symbol names on AIX
 
 	FILTER="_(descriptor|equal|hash)"
-	tests=$($NM -p -v _gotest_.o $xofile | egrep " $text .*\."$pattern'$' | grep -v '\..*\..*\.' | fgrep -v '$' | egrep -v "$FILTER" | fgrep -v ' __go_' | sed 's/.* //' | $symtogo | sed "s/^\.//" )
+	tests=$($NM -p -v _gotest_.o $xofile | egrep " $text .*\."$pattern'$' | grep -v '\..*\..*\.' | fgrep -v '$' | egrep -v "$FILTER" | /usr/bin/awk '{n0=match($0,"_gc"); n1=match($0, ".go_internal_gc"); if(n0==0||n1!=0) print $0}' | fgrep -v ' __go_' | sed 's/.* //' | $symtogo | sed "s/^\.//" )
         echo $tests > /tmp/testsgo
 	if [ "x$tests" = x ]; then
 		echo 'gotest: warning: no tests matching '$pattern in _gotest_.o $xofile 1>&2
@@ -551,11 +551,11 @@
 	fi
 	# benchmarks are named BenchmarkFoo.
 	pattern='Benchmark([^a-z].*)?'
-	benchmarks=$($NM -p -v _gotest_.o $xofile | egrep " $text .*\."$pattern'$' | grep -v '\..*\..*\.' | fgrep -v '$' | egrep -v "$FILTER" | fgrep -v ' __go_' | sed 's/.* //' | $symtogo | sed "s/^\.//" )
+	benchmarks=$($NM -p -v _gotest_.o $xofile | egrep " $text .*\."$pattern'$' | grep -v '\..*\..*\.' | fgrep -v '$' | egrep -v "$FILTER" | /usr/bin/awk '{n0=match($0,"_gc"); n1=match($0, ".go_internal_gc"); if(n0==0||n1!=0) print $0}' | fgrep -v ' __go_' | sed 's/.* //' | $symtogo | sed "s/^\.//" )
 
 	# examples are named ExampleFoo
 	pattern='Example([^a-z].*)?'
-	examples=$($NM -p -v _gotest_.o $xofile | egrep " $text .*\."$pattern'$' | grep -v '\..*\..*\.' | fgrep -v '$' | egrep -v "$FILTER" | fgrep -v ' __go_' | sed 's/.* //' | $symtogo | sed "s/^\.//" )
+	examples=$($NM -p -v _gotest_.o $xofile | egrep " $text .*\."$pattern'$' | grep -v '\..*\..*\.' | fgrep -v '$' | egrep -v "$FILTER" | /usr/bin/awk '{n0=match($0,"_gc"); n1=match($0, ".go _internal_gc"); if(n0==0||n1!=0) print $0}' | fgrep -v ' __go_' | sed 's/.* //' | $symtogo | sed "s/^\.//" )
 
 	# package spec
 	echo 'package main'
@@ -642,7 +642,7 @@
 		fi
 		./a.out -test.short -test.timeout=${timeout}s "$@" &
 		pid=$!
-		(sleep `expr $timeout + 10`
+		(sleep `/usr/bin/expr $timeout + 10`
 		    echo > gotest-timeout
 		    echo "timed out in gotest" 1>&2
 		    kill -9 $pid) &
