Index: contrib/config-list.mk
===================================================================
--- a/contrib/config-list.mk	(revision 270913)
+++ b/contrib/config-list.mk	(working copy)
@@ -123,7 +123,7 @@ $(LIST): make-log-dir
 		TGT=`echo $@ | awk 'BEGIN { FS = "OPT" }; { print $$1 }'` &&			\
 		TGT=`$(GCC_SRC_DIR)/config.sub $$TGT` &&					\
 		case $$TGT in									\
-			*-*-darwin* | *-*-cygwin* | *-*-mingw* | *-*-aix*)			\
+			*-*-darwin* | *-*-cygwin* | *-*-mingw*)			\
 				ADDITIONAL_LANGUAGES="";					\
 				;;								\
 			*)									\
