diff --git a/libgo/go/os/user/decls_aix.go b/libgo/go/os/user/decls_aix.go
--- a/libgo/go/os/user/decls_aix.go
+++ b/libgo/go/os/user/decls_aix.go
@@ -8,16 +8,16 @@
 
 // Declarations for the libc functions on AIX.
 
-//extern getpwnam_r
+//extern _posix_getpwnam_r
 func libc_getpwnam_r(name *byte, pwd *syscall.Passwd, buf *byte, buflen syscall.Size_t, result **syscall.Passwd) int
 
-//extern getpwuid_r
+//extern _posix_getpwuid_r
 func libc_getpwuid_r(uid syscall.Uid_t, pwd *syscall.Passwd, buf *byte, buflen syscall.Size_t, result **syscall.Passwd) int
 
-//extern getgrnam_r
+//extern _posix_getgrnam_r
 func libc_getgrnam_r(name *byte, grp *syscall.Group, buf *byte, buflen syscall.Size_t, result **syscall.Group) int
 
-//extern getgrgid_r
+//extern _posix_getgrgid_r
 func libc_getgrgid_r(gid syscall.Gid_t, grp *syscall.Group, buf *byte, buflen syscall.Size_t, result **syscall.Group) int
 
 //extern getgrset
