--- ./libgo/testsuite/gotest.ORIGIN	2017-06-14 15:26:12 +0200
+++ ./libgo/testsuite/gotest	2017-06-14 15:34:00 +0200
@@ -8,6 +8,8 @@
 # tests.
 # If files are named on the command line, use them instead of *_test.go.
 
+#set -x
+
 # Makes egrep,grep work better in general if we put them
 # in ordinary C mode instead of what the current language is.
 LANG=C
@@ -33,13 +35,15 @@
 goos=""
 pkgfiles=""
 loop=true
+keep=true
 keep=false
 pkgpath=
 prefix=
 dejagnu=no
-timeout=240
+timeout=500
 testname=""
 bench=""
+trace=true
 trace=false
 while $loop; do
 	case "x$1" in
@@ -86,6 +90,11 @@
 		;;
 	x--pkgpath=*)
 		pkgpath=`echo $1 | sed -e 's/^--pkgpath=//'`
+#		echo "Test: $pkgpath"
+#		if [ "$pkgpath" != "net/http/httputil" ]
+#		then
+#		  exit 0
+#		fi
 		shift
 		;;
 	x--prefix)
@@ -638,9 +647,9 @@
 	set +e
 	if test "$bench" = ""; then
 		if test "$trace" = "true"; then
-		    echo ./a.out -test.short -test.timeout=${timeout}s "$@"
+		    echo ./a.out -test.v -test.timeout=${timeout}s "$@"
 		fi
-		./a.out -test.short -test.timeout=${timeout}s "$@" &
+		./a.out -test.v -test.timeout=${timeout}s "$@" &
 		pid=$!
 		(sleep `/usr/bin/expr $timeout + 10`
 		    echo > gotest-timeout
