Index: gcc-4.4.0/libjava/Makefile.am
===================================================================
--- gcc-4.4.0.orig/libjava/Makefile.am	2009-07-10 15:19:48.000000000 +0200
+++ gcc-4.4.0/libjava/Makefile.am	2009-07-10 15:22:17.000000000 +0200
@@ -276,7 +276,7 @@
 # Include THREADLIBS here to ensure that the correct version of
 # certain linuxthread functions get linked:
 ## The mysterious backslash in the grep pattern is consumed by make.
-libgcj_la_LDFLAGS = -rpath $(toolexeclibdir) $(THREADLDFLAGS) $(extra_ldflags) $(THREADLIBS) \
+libgcj_la_LDFLAGS = -Wl,-bbigtoc -rpath $(toolexeclibdir) $(THREADLDFLAGS) $(extra_ldflags) $(THREADLIBS) \
 	$(LIBLTDL) $(SYS_ZLIBS) \
 	-version-info `grep -v '^\#' $(srcdir)/libtool-version` \
 	$(LIBGCJ_LD_SYMBOLIC_FUNCTIONS)
@@ -369,10 +369,10 @@
 	$(libgcj_bc_la_OBJECTS) $(libgcj_bc_la_LIBADD) $(LIBS) || exit; \
 	rm .libs/libgcj_bc.so; \
 	mv .libs/libgcj_bc.so.1.0.0 .libs/libgcj_bc.so; \
-	$(libgcj_bc_dummy_LINK) -xc /dev/null -Wl,-soname,libgcj_bc.so.1 \
-	-o .libs/libgcj_bc.so.1.0.0 -lgcj || exit; \
-	rm .libs/libgcj_bc.so.1; \
-	$(LN_S) libgcj_bc.so.1.0.0 .libs/libgcj_bc.so.1
+	$(libgcj_bc_dummy_LINK) -xc /dev/null -Wl,-soname,.libs/libgcj_bc.so.1 \
+	-o .libs/libgcj_bc.so.1.0.0 -lgcj || exit;
+#	rm .libs/libgcj_bc.so.1; \
+#	$(LN_S) libgcj_bc.so.1.0.0 .libs/libgcj_bc.so.1
 
 ## Note that property_files is defined in sources.am.
 propertyo_files = $(patsubst classpath/resource/%,%,$(addsuffix .lo,$(property_files)))
@@ -499,11 +499,11 @@
 	@echo Installing dummy lib libgcj_bc.so.1.0.0; \
 	rm $(DESTDIR)$(toolexeclibdir)/libgcj_bc.so; \
 	mv $(DESTDIR)$(toolexeclibdir)/libgcj_bc.so.1.0.0 $(DESTDIR)$(toolexeclibdir)/libgcj_bc.so; \
-	$(libgcj_bc_dummy_LINK) -xc /dev/null -Wl,-soname,libgcj_bc.so.1 \
-	-o $(DESTDIR)$(toolexeclibdir)/libgcj_bc.so.1.0.0 -lgcj || exit; \
-	rm $(DESTDIR)$(toolexeclibdir)/libgcj_bc.so.1; \
-	$(LN_S) libgcj_bc.so.1.0.0 $(DESTDIR)$(toolexeclibdir)/libgcj_bc.so.1; \
-	rm $(DESTDIR)$(toolexeclibdir)/libgcj_bc.la;
+	$(libgcj_bc_dummy_LINK) -xc /dev/null -Wl,-soname,.libs/libgcj_bc.so.1 \
+	-o $(DESTDIR)$(toolexeclibdir)/libgcj_bc.so.1.0.0 -lgcj || exit;
+#	rm $(DESTDIR)$(toolexeclibdir)/libgcj_bc.so.1; \
+#	$(LN_S) libgcj_bc.so.1.0.0 $(DESTDIR)$(toolexeclibdir)/libgcj_bc.so.1; \
+#	rm $(DESTDIR)$(toolexeclibdir)/libgcj_bc.la;
 endif
 if BUILD_ECJ1
 ## Undo any prepending of the target prefix and transform
@@ -834,7 +834,7 @@
 ## searched before the build-tree ones, and we'd get errors because of
 ## different libraries with the same SONAME from picky linkers such as
 ## Solaris'.  FIXME: should be _libs on some systems.
-gcj_dbtool_LDADD = gnu/gcj/tools/gcj_dbtool.lo -L$(here)/.libs libgcj.la
+gcj_dbtool_LDADD = gnu/gcj/tools/gcj_dbtool.lo -lgcj-tools -lgcj_bc -L$(here)/.libs libgcj.la
 ## Depend on the spec file to make sure it is up to date before
 ## linking this program.
 gcj_dbtool_DEPENDENCIES = gnu/gcj/tools/gcj_dbtool.lo libgcj.la libgcj.spec
Index: gcc-4.4.0/libjava/Makefile.in
===================================================================
--- gcc-4.4.0.orig/libjava/Makefile.in	2009-07-10 15:19:57.000000000 +0200
+++ gcc-4.4.0/libjava/Makefile.in	2009-07-10 16:39:38.000000000 +0200
@@ -1007,7 +1007,7 @@
 
 # Include THREADLIBS here to ensure that the correct version of
 # certain linuxthread functions get linked:
-libgcj_la_LDFLAGS = -rpath $(toolexeclibdir) $(THREADLDFLAGS) $(extra_ldflags) $(THREADLIBS) \
+libgcj_la_LDFLAGS = -Wl,-bbigtoc -rpath $(toolexeclibdir) $(THREADLDFLAGS) $(extra_ldflags) $(THREADLIBS) \
 	$(LIBLTDL) $(SYS_ZLIBS) \
 	-version-info `grep -v '^\#' $(srcdir)/libtool-version` \
 	$(LIBGCJ_LD_SYMBOLIC_FUNCTIONS)
@@ -8332,7 +8332,7 @@
 	-rpath $(toolexeclibdir) -shared-libgcc $(THREADLDFLAGS) $(extra_ldflags)
 
 gcj_dbtool_LINK = $(GCJLINK)
-gcj_dbtool_LDADD = gnu/gcj/tools/gcj_dbtool.lo -L$(here)/.libs libgcj.la
+gcj_dbtool_LDADD = gnu/gcj/tools/gcj_dbtool.lo -lgcj-tools -lgcj_bc -L$(here)/.libs libgcj.la
 gcj_dbtool_DEPENDENCIES = gnu/gcj/tools/gcj_dbtool.lo libgcj.la libgcj.spec
 gc_analyze_SOURCES = 
 gc_analyze_LDFLAGS = --main=gnu.gcj.tools.gc_analyze.MemoryAnalyze \
@@ -10517,10 +10517,10 @@
 	$(libgcj_bc_la_OBJECTS) $(libgcj_bc_la_LIBADD) $(LIBS) || exit; \
 	rm .libs/libgcj_bc.so; \
 	mv .libs/libgcj_bc.so.1.0.0 .libs/libgcj_bc.so; \
-	$(libgcj_bc_dummy_LINK) -xc /dev/null -Wl,-soname,libgcj_bc.so.1 \
-	-o .libs/libgcj_bc.so.1.0.0 -lgcj || exit; \
-	rm .libs/libgcj_bc.so.1; \
-	$(LN_S) libgcj_bc.so.1.0.0 .libs/libgcj_bc.so.1
+	$(libgcj_bc_dummy_LINK) -xc /dev/null -Wl,-soname,.libs/libgcj_bc.so.1 \
+	-o .libs/libgcj_bc.so.1.0.0 -lgcj || exit;
+#	rm .libs/libgcj_bc.so.1; \
+#	$(LN_S) libgcj_bc.so.1.0.0 .libs/libgcj_bc.so.1
 
 $(propertyo_files): %.lo: classpath/resource/%
 	$(mkinstalldirs) `dirname $@`; \
@@ -12363,10 +12363,10 @@
 @USE_LIBGCJ_BC_TRUE@	@echo Installing dummy lib libgcj_bc.so.1.0.0; \
 @USE_LIBGCJ_BC_TRUE@	rm $(DESTDIR)$(toolexeclibdir)/libgcj_bc.so; \
 @USE_LIBGCJ_BC_TRUE@	mv $(DESTDIR)$(toolexeclibdir)/libgcj_bc.so.1.0.0 $(DESTDIR)$(toolexeclibdir)/libgcj_bc.so; \
-@USE_LIBGCJ_BC_TRUE@	$(libgcj_bc_dummy_LINK) -xc /dev/null -Wl,-soname,libgcj_bc.so.1 \
-@USE_LIBGCJ_BC_TRUE@	-o $(DESTDIR)$(toolexeclibdir)/libgcj_bc.so.1.0.0 -lgcj || exit; \
-@USE_LIBGCJ_BC_TRUE@	rm $(DESTDIR)$(toolexeclibdir)/libgcj_bc.so.1; \
-@USE_LIBGCJ_BC_TRUE@	$(LN_S) libgcj_bc.so.1.0.0 $(DESTDIR)$(toolexeclibdir)/libgcj_bc.so.1; \
+@USE_LIBGCJ_BC_TRUE@	$(libgcj_bc_dummy_LINK) -xc /dev/null -Wl,-soname,.libs/libgcj_bc.so.1 \
+@USE_LIBGCJ_BC_TRUE@	-o $(DESTDIR)$(toolexeclibdir)/libgcj_bc.so.1.0.0 -lgcj || exit;
+@USE_LIBGCJ_BC_TRUE@#	rm $(DESTDIR)$(toolexeclibdir)/libgcj_bc.so.1; \
+@USE_LIBGCJ_BC_TRUE@#	$(LN_S) libgcj_bc.so.1.0.0 $(DESTDIR)$(toolexeclibdir)/libgcj_bc.so.1;
 @USE_LIBGCJ_BC_TRUE@	rm $(DESTDIR)$(toolexeclibdir)/libgcj_bc.la;
 @BUILD_ECJ1_TRUE@	mv $(DESTDIR)$(libexecsubdir)/`echo ecjx | sed 's,^.*/,,;$(transform);s/$$/$(EXEEXT)/'` $(DESTDIR)$(libexecsubdir)/ecj1$(host_exeext)
 install-data-local:
Index: gcc-4.4.0/libjava/configure.host
===================================================================
--- gcc-4.4.0.orig/libjava/configure.host	2009-07-10 15:20:08.000000000 +0200
+++ gcc-4.4.0/libjava/configure.host	2009-07-10 15:22:35.000000000 +0200
@@ -368,6 +368,9 @@
   *linux*|*-kfreebsd*-gnu|*-gnu*)
     use_libgcj_bc=yes
   ;;
+  *aix*)
+    use_libgcj_bc=yes
+  ;;
   *)
     use_libgcj_bc=no
   ;;
