Index: libgo/go/cmd/go/internal/work/gccgo.go
===================================================================
--- a/libgo/go/cmd/go/internal/work/gccgo.go	(revision 272180)
+++ b/libgo/go/cmd/go/internal/work/gccgo.go	(working copy)
@@ -209,7 +209,7 @@ func (tools gccgoToolchain) pack(b *Builder, a *Ac
 	}
 	absAfile := mkAbs(objdir, afile)
 	// Try with D modifier first, then without if that fails.
-	if b.run(a, p.Dir, p.ImportPath, nil, tools.ar(), arArgs, "rcD", absAfile, absOfiles) != nil {
+	if _, err := b.runOut(p.Dir, nil, tools.ar(), arArgs, "rcD", absAfile, absOfiles); err != nil {
 		return b.run(a, p.Dir, p.ImportPath, nil, tools.ar(), arArgs, "rc", absAfile, absOfiles)
 	}
 	return nil
