diff --git a/libgo/go/os/file_posix.go b/libgo/go/os/file_posix.go
--- a/libgo/go/os/file_posix.go
+++ b/libgo/go/os/file_posix.go
@@ -19,6 +19,10 @@
 	for len := 128; ; len *= 2 {
 		b := make([]byte, len)
 		n, e := fixCount(syscall.Readlink(fixLongPath(name), b))
+		// buffer too small
+		if e == syscall.ERANGE {
+			continue
+		}
 		if e != nil {
 			return "", &PathError{"readlink", name, e}
 		}
