Index: bzip2-1.0.2/Makefile-libbz2_so
diff -c bzip2-1.0.2/Makefile-libbz2_so:1.1.1.1 bzip2-1.0.1/Makefile-libbz2_so:1.5
*** bzip2-1.0.2/Makefile-libbz2_so:1.1.1.1	Sat Aug 12 22:13:55 2000
--- bzip2-1.0.2/Makefile-libbz2_so	Sun Mar 25 19:07:22 2001
***************
*** 20,28 ****
  
  all: $(OBJS)
  	$(CC) -shared -Wl,-soname -Wl,libbz2.so.1.0 -o libbz2.so.1.0.2 $(OBJS)
! 	$(CC) $(CFLAGS) -o bzip2-shared bzip2.c libbz2.so.1.0.2
! 	rm -f libbz2.so.1.0
  	ln -s libbz2.so.1.0.2 libbz2.so.1.0
  
  clean: 
  	rm -f $(OBJS) bzip2.o libbz2.so.1.0.2 libbz2.so.1.0 bzip2-shared
--- 20,39 ----
  
  all: $(OBJS)
  	$(CC) -shared -Wl,-soname -Wl,libbz2.so.1.0 -o libbz2.so.1.0.2 $(OBJS)
! 	rm -f libbz2.so.1.0 libz2.so.1 libbz2.so
  	ln -s libbz2.so.1.0.2 libbz2.so.1.0
+ 	ln -s libbz2.so.1.0.2 libbz2.so.1
+ 	ln -s libbz2.so.1.0.2 libbz2.so
+ 	$(CC) $(CFLAGS) -c bzip2.c
+ 	$(CC) $(CFLAGS) -o bzip2-shared bzip2.o libbz2.so.1.0.2
+ 
+ all_ppc: $(OBJS)
+ 	$(CC) -shared -o libbz2.so.1.0.2 $(OBJS)
+ 	rm -f libbz2.so.1.0 libbz2.a
+ 	ln -s libbz2.so.1.0.2 libbz2.so.1
+ 	ar -qv libbz2.a libbz2.so.1
+ 	$(CC) $(CFLAGS) -c bzip2.c
+ 	$(CC) $(CFLAGS) -o bzip2-shared bzip2.o libbz2.a
  
  clean: 
  	rm -f $(OBJS) bzip2.o libbz2.so.1.0.2 libbz2.so.1.0 bzip2-shared
Index: bzip2-1.0.2/Makefile-libbz2_so-vac
diff -c /dev/null bzip2-1.0.2/Makefile-libbz2_so-vac:1.1
*** /dev/null	Sun Mar 25 19:07:28 2001
--- bzip2-1.0.2/Makefile-libbz2_so-vac	Tue Oct 10 17:14:11 2000
***************
*** 0 ****
--- 1,44 ----
+ 
+ # This Makefile builds a shared version of the library, 
+ # libbz2.so.1.0.2, at least on ia64-AIX with xlC 5.5.  
+ # Please see the README file for some 
+ # important info about building the library like this.
+ 
+ SHELL=/bin/sh
+ CC=xlc
+ BIGFILES=-D_FILE_OFFSET_BITS=64
+ CFLAGS=-O2 $(BIGFILES)
+ 
+ OBJS= blocksort.o  \
+       huffman.o    \
+       crctable.o   \
+       randtable.o  \
+       compress.o   \
+       decompress.o \
+       bzlib.o
+ 
+ all: $(OBJS)
+ # This may need to be changed from -G to -qmkshrobj.
+ 	$(CC) -G -o libbz2.so.1.0.2 $(OBJS)
+ 	$(CC) $(CFLAGS) -o bzip2-shared bzip2.c libbz2.so.1.0.2
+ 	rm -f libbz2.so.1.0
+ 	ln -s libbz2.so.1.0.2 libbz2.so.1.0
+ 	ln -s libbz2.so.1.0.2 libbz2.so
+ 
+ clean: 
+ 	rm -f $(OBJS) bzip2.o libbz2.so.1.0.2 libbz2.so.1.0 bzip2-shared
+ 
+ blocksort.o: blocksort.c
+ 	$(CC) $(CFLAGS) -c blocksort.c
+ huffman.o: huffman.c
+ 	$(CC) $(CFLAGS) -c huffman.c
+ crctable.o: crctable.c
+ 	$(CC) $(CFLAGS) -c crctable.c
+ randtable.o: randtable.c
+ 	$(CC) $(CFLAGS) -c randtable.c
+ compress.o: compress.c
+ 	$(CC) $(CFLAGS) -c compress.c
+ decompress.o: decompress.c
+ 	$(CC) $(CFLAGS) -c decompress.c
+ bzlib.o: bzlib.c
+ 	$(CC) $(CFLAGS) -c bzlib.c
Index: bzip2-1.0.2/Makefile-libbz2_so.nongcc
diff -c /dev/null bzip2-1.0.2/Makefile-libbz2_so.nongcc:1.7
*** /dev/null	Sun Mar 25 19:07:28 2001
--- bzip2-1.0.2/Makefile-libbz2_so.nongcc	Sun Mar 25 16:28:28 2001
***************
*** 0 ****
--- 1,53 ----
+ 
+ # This Makefile builds a shared version of the library, 
+ # libbz2.so.1.0.2, with soname libbz2.so.1.0 on AIX without gcc
+ # Please see the README file for some 
+ # important info about building the library like this.
+ 
+ SHELL=/bin/sh
+ CC=cc
+ BIGFILES=-D_FILE_OFFSET_BITS=64
+ CFLAGS=-O2 
+ 
+ OBJS= blocksort.o  \
+       huffman.o    \
+       crctable.o   \
+       randtable.o  \
+       compress.o   \
+       decompress.o \
+       bzlib.o
+ 
+ all: $(OBJS)
+ 	$(CC) -G -Wl,-hlibbz2.so.1 -o libbz2.so.1.0.2 $(OBJS)
+ 	rm -f libbz2.so.1.0 libbz2.so.1 libbz2.so
+ 	ln -s libbz2.so.1.0.2 libbz2.so.1.0
+ 	ln -s libbz2.so.1.0.2 libbz2.so.1
+ 	ln -s libbz2.so.1.0.2 libbz2.so
+ 	$(CC) $(CFLAGS) -c bzip2.o
+ 	$(CC) $(CFLAGS) -o bzip2-shared bzip2.o libbz2.so
+ 
+ all_ppc: $(OBJS)
+ 	$(CC) -G -o libbz2.so.1.0.2 $(OBJS)
+ 	rm -f libbz2.so.1.0 libbz2.a
+ 	ln -s libbz2.so.1.0.2 libbz2.so.1
+ 	ar -qv libbz2.a libbz2.so.1
+ 	$(CC) $(CFLAGS) -c bzip2.o
+ 	$(CC) $(CFLAGS) -o bzip2-shared bzip2.o libbz2.a
+ 
+ clean: 
+ 	rm -f $(OBJS) bzip2.o libnz2.a libbz2.so.1.0.2 libbz2.so.1.0 bzip2-shared
+ 
+ blocksort.o: blocksort.c
+ 	$(CC) $(CFLAGS) -c blocksort.c
+ huffman.o: huffman.c
+ 	$(CC) $(CFLAGS) -c huffman.c
+ crctable.o: crctable.c
+ 	$(CC) $(CFLAGS) -c crctable.c
+ randtable.o: randtable.c
+ 	$(CC) $(CFLAGS) -c randtable.c
+ compress.o: compress.c
+ 	$(CC) $(CFLAGS) -c compress.c
+ decompress.o: decompress.c
+ 	$(CC) $(CFLAGS) -c decompress.c
+ bzlib.o: bzlib.c
+ 	$(CC) $(CFLAGS) -c bzlib.c
