--- Python-3.5.1-orig/Lib/test/test_fileio.py	2015-12-07 02:39:09 +0100
+++ Python-3.5.1-aix/Lib/test/test_fileio.py	2016-06-22 14:13:38 +0200
@@ -377,7 +377,8 @@
                     self.assertEqual(f.writable(), True)
                     if sys.platform != "darwin" and \
                        'bsd' not in sys.platform and \
-                       not sys.platform.startswith('sunos'):
+                       not sys.platform.startswith('sunos') and \
+                       not sys.platform.startswith('aix'):
                         # Somehow /dev/tty appears seekable on some BSDs
                         self.assertEqual(f.seekable(), False)
                     self.assertEqual(f.isatty(), True)
