--- ./storage/innobase/include/univ.i.orig	2020-07-02 15:57:55 +0200
+++ ./storage/innobase/include/univ.i	2020-07-02 15:58:37 +0200
@@ -418,10 +418,10 @@
 # define UINT64PFx	"%016llx"
 #else
 /* Use the integer types and formatting strings defined in the C99 standard. */
-# define UINT32PF	"%" PRIu32
-# define INT64PF	"%" PRId64
-# define UINT64scan	PRIu64
-# define UINT64PFx	"%016" PRIx64
+# define UINT32PF	"%u"
+# define INT64PF	"%lld"
+# define UINT64scan	"llu"
+# define UINT64PFx	"%016llx"
 #endif
 
 #ifdef UNIV_INNOCHECKSUM
