--- ./support-files/CMakeLists.txt.orig	2018-10-25 13:47:47 -0500
+++ ./support-files/CMakeLists.txt	2018-10-25 13:59:28 -0500
@@ -65,19 +65,19 @@
     MARK_AS_ADVANCED(CHECKMODULE SEMODULE_PACKAGE)
 
     # Build pp files in policy/selinux
-    IF(CHECKMODULE AND SEMODULE_PACKAGE)
-      FOREACH(pol mariadb)
-        SET(src ${CMAKE_CURRENT_SOURCE_DIR}/policy/selinux/${pol}.te)
-        SET(tmp ${CMAKE_CURRENT_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/${pol}-pp.dir/${pol}.mod)
-        SET(out ${CMAKE_CURRENT_BINARY_DIR}/${pol}.pp)
-        ADD_CUSTOM_COMMAND(OUTPUT ${out}
-          COMMAND ${CHECKMODULE} -M -m ${src} -o ${tmp}
-          COMMAND ${SEMODULE_PACKAGE} -m ${tmp} -o ${out}
-        DEPENDS ${src})
-        ADD_CUSTOM_TARGET(${pol}-pp ALL DEPENDS ${out})
-        INSTALL(FILES ${out} DESTINATION ${inst_location}/policy/selinux COMPONENT SupportFiles)
-      ENDFOREACH()
-    ENDIF()
+#    IF(CHECKMODULE AND SEMODULE_PACKAGE)
+#      FOREACH(pol mariadb)
+#        SET(src ${CMAKE_CURRENT_SOURCE_DIR}/policy/selinux/${pol}.te)
+#        SET(tmp ${CMAKE_CURRENT_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/${pol}-pp.dir/${pol}.mod)
+#        SET(out ${CMAKE_CURRENT_BINARY_DIR}/${pol}.pp)
+#        ADD_CUSTOM_COMMAND(OUTPUT ${out}
+#          COMMAND ${CHECKMODULE} -M -m ${src} -o ${tmp}
+#          COMMAND ${SEMODULE_PACKAGE} -m ${tmp} -o ${out}
+#        DEPENDS ${src})
+#        ADD_CUSTOM_TARGET(${pol}-pp ALL DEPENDS ${out})
+#        INSTALL(FILES ${out} DESTINATION ${inst_location}/policy/selinux COMPONENT SupportFiles)
+#      ENDFOREACH()
+#    ENDIF()
   ENDIF()
 
   CONFIGURE_FILE(mariadb.pc.in ${CMAKE_CURRENT_BINARY_DIR}/mariadb.pc @ONLY)
