--- ./sql/lex.h.orig	2018-10-17 08:39:55 -0500
+++ ./sql/lex.h	2018-10-17 08:40:19 -0500
@@ -497,7 +497,7 @@
   { "QUICK",	        SYM(QUICK)},
   { "RAISE",            SYM(RAISE_SYM)},
   { "RANGE",            SYM(RANGE_SYM)},
-  { "RAW",              SYM(RAW)},
+  { "RAW",              SYM(RAW_VALUE)},
   { "READ",		SYM(READ_SYM)},
   { "READ_ONLY",	SYM(READ_ONLY_SYM)},
   { "READ_WRITE",	SYM(READ_WRITE_SYM)},
