--- ./libmariadb/include/ma_global.h.orig	2018-10-17 14:23:40 -0500
+++ ./libmariadb/include/ma_global.h	2018-10-17 14:21:49 -0500
@@ -615,7 +615,7 @@
 typedef char	*gptr;		/* Generic pointer */
 #endif
 #ifndef HAVE_INT_8_16_32
-typedef char	int8;		/* Signed integer >= 8	bits */
+typedef signed char	int8;		/* Signed integer >= 8	bits */
 typedef short	int16;		/* Signed integer >= 16 bits */
 #endif
 #ifndef HAVE_UCHAR
