--- ./cmake/build_configurations/mysql_release.cmake.orig	2019-04-18 17:00:16.164000000 +0200
+++ ./cmake/build_configurations/mysql_release.cmake	2019-04-25 13:48:20.164000000 +0200
@@ -92,6 +92,10 @@
   SET(INSTALL_MYSQLTESTDIR "" CACHE STRING "")
   SET(INSTALL_SQLBENCHDIR  "" CACHE STRING "")
   SET(INSTALL_SUPPORTFILESDIR ""  CACHE STRING "")
+ELSEIF(AIX)
+  SET(WITH_SSL system CACHE STRING "")
+  SET(WITH_ZLIB system CACHE STRING "")
+  SET(WITH_LIBARCHIVE STATIC CACHE STRING "")
 ELSEIF(RPM)
   SET(WITH_SSL system CACHE STRING "")
   SET(WITH_ZLIB system CACHE STRING "")
@@ -266,4 +270,19 @@
       ENDIF()
     ENDIF()
   ENDIF()
+
+  # AIX flags
+  IF(AIX)
+    IF(CMAKE_COMPILER_IS_GNUCC)
+     SET(COMMON_C_FLAGS               "-fno-omit-frame-pointer -fno-strict-aliasing  -Wno-uninitialized")
+     SET(CMAKE_C_FLAGS_DEBUG          "-g -O0 ${COMMON_C_FLAGS}")
+     SET(CMAKE_C_FLAGS_RELEASE        "-O3    ${COMMON_C_FLAGS}") # -g -O0
+   ENDIF()
+   IF(CMAKE_COMPILER_IS_GNUCXX)
+     SET(COMMON_CXX_FLAGS               "-fno-omit-frame-pointer -fno-strict-aliasing -Wno-uninitialized")
+     SET(CMAKE_CXX_FLAGS_DEBUG          "-g -O0 ${COMMON_CXX_FLAGS}")
+     SET(CMAKE_CXX_FLAGS_RELEASE        "-O3    ${COMMON_CXX_FLAGS}") # -g -O0
+   ENDIF()
+ ENDIF()
+
 ENDIF()
