--- ./libmariadb/include/ma_global.h.orig	2019-06-27 11:38:47.000000000 +0200
+++ ./libmariadb/include/ma_global.h	2019-06-27 11:38:47.000000000 +0200
@@ -91,7 +91,7 @@
 
 /* Go around some bugs in different OS and compilers */
 #ifdef _AIX			/* By soren@t.dk */
-#define _H_STRINGS
+/* #define _H_STRINGS
 #define _SYS_STREAM_H
 #define _AIX32_CURSES
 #define ulonglong2double(A) my_ulonglong2double(A)
@@ -102,7 +102,7 @@
 double my_ulonglong2double(unsigned long long A);
 #ifdef	__cplusplus
 }
-#endif
+#endif */
 #endif /* _AIX */
 
 #ifdef HAVE_BROKEN_SNPRINTF	/* HPUX 10.20 don't have this defined */
--- ./storage/innobase/CMakeLists.txt.orig	2019-06-27 12:13:55.000000000 +0200
+++ ./storage/innobase/CMakeLists.txt	2019-06-27 12:13:55.000000000 +0200
@@ -155,7 +155,6 @@
 	${ZLIB_LIBRARY}
 	${CRC32_LIBRARY}
 	${NUMA_LIBRARY}
-	${LIBSYSTEMD}
 	${LINKER_SCRIPT})
 
 IF(NOT TARGET innobase)
--- ./libmysqld/CMakeLists.txt.orig	2019-06-27 11:46:10.000000000 +0200
+++ ./libmysqld/CMakeLists.txt	2019-07-29 11:53:58.020807791 +0200
@@ -72,7 +72,7 @@
            ../sql/sql_admin.cc ../sql/sql_truncate.cc ../sql/sql_reload.cc
            ../sql/sql_lex.cc ../sql/keycaches.cc
            ../sql/sql_list.cc ../sql/sql_load.cc ../sql/sql_locale.cc 
-           ../sql/sql_binlog.cc ../sql/sql_manager.cc
+           ../sql/sql_manager.cc
            ../sql/sql_parse.cc ../sql/sql_bootstrap.cc
            ../sql/sql_partition.cc ../sql/sql_plugin.cc 
            ../sql/debug_sync.cc ../sql/opt_table_elimination.cc
@@ -126,7 +126,6 @@
            ${MYSYS_LIBWRAP_SOURCE}
 )
 
-
 ADD_CONVENIENCE_LIBRARY(sql_embedded ${SQL_EMBEDDED_SOURCES})
 DTRACE_INSTRUMENT(sql_embedded)
 ADD_DEPENDENCIES(sql_embedded GenError GenServerSource)
@@ -137,6 +136,10 @@
   SET(MYSQLSERVER_OUTPUT_NAME mysqlserver)
   SET(COMPONENT_MYSQLSERVER "Embedded")
   SET(COMPONENT_LIBMYSQLD "Embedded")
+ELSEIF(AIX)
+  SET(MYSQLSERVER_OUTPUT_NAME mariadbd_server)
+  SET(COMPONENT_MYSQLSERVER "Embedded")
+  SET(COMPONENT_LIBMYSQLD "Embedded")
 ELSE()
   SET(MYSQLSERVER_OUTPUT_NAME mariadbd)
   SET(COMPONENT_MYSQLSERVER "Development")
@@ -165,7 +168,7 @@
 
 MERGE_LIBRARIES(mysqlserver STATIC ${EMBEDDED_LIBS} 
   OUTPUT_NAME ${MYSQLSERVER_OUTPUT_NAME} COMPONENT ${COMPONENT_MYSQLSERVER})
-IF(UNIX)
+IF(UNIX AND NOT AIX)
   INSTALL_SYMLINK(libmysqld.a mysqlserver ${INSTALL_LIBDIR} ${COMPONENT_MYSQLSERVER})
 ENDIF()
 INSTALL(FILES embedded_priv.h DESTINATION ${INSTALL_INCLUDEDIR}/server/private COMPONENT  ${COMPONENT_MYSQLSERVER})
@@ -441,7 +444,11 @@
     SET_TARGET_PROPERTIES(libmysqld PROPERTIES 
       OUTPUT_NAME mariadbd
       SOVERSION "${SHARED_LIB_MAJOR_VERSION}")
-    INSTALL_SYMLINK(libmysqld.so libmysqld ${INSTALL_LIBDIR} ${COMPONENT_LIBMYSQLD})
+    IF(AIX)
+      INSTALL_SYMLINK(libmysqld.a libmysqld ${INSTALL_LIBDIR} ${COMPONENT_LIBMYSQLD})
+    ELSE()
+      INSTALL_SYMLINK(libmysqld.so libmysqld ${INSTALL_LIBDIR} ${COMPONENT_LIBMYSQLD})
+    ENDIF()
     # Clean direct output flags, as 2 targets have the same base name
     # libmysqld
     SET_TARGET_PROPERTIES(libmysqld PROPERTIES CLEAN_DIRECT_OUTPUT 1)
--- ./sql/CMakeLists.txt.orig	2019-06-27 11:46:43.000000000 +0200
+++ ./sql/CMakeLists.txt	2019-06-28 16:24:01.000000000 +0200
@@ -287,6 +287,24 @@
 
 MYSQL_ADD_EXECUTABLE(mysqld ${MYSQLD_SOURCE} DESTINATION ${INSTALL_SBINDIR} COMPONENT Server)
 
+IF(AIX)
+  add_custom_target(mysql_symbols
+      COMMAND printf "#!\\n" > /tmp/tmp.exp
+      COMMAND CreateExportList /tmp/libmariadbd_tmp.exp libsql.a ../mysys/libmysys.a ../dbug/libdbug.a ../strings/libstrings.a ../vio/libvio.a -X${OBJECT_MODE}
+      COMMAND cat /tmp/tmp.exp /tmp/libmariadbd_tmp.exp > /tmp/libmariadbd.exp
+      COMMAND rm /tmp/libmariadbd_tmp.exp /tmp/tmp.exp
+    VERBATIM
+  )
+  ADD_DEPENDENCIES(mysql_symbols sql mysys dbug strings vio)
+# COMMAND printf "#! libsql.a\\n#! libmysys.a\\n#! libdbug.a\\n#! libstrings.a\\n#! libvio.a\\n" > /tmp/tmp.exp
+#   add_custom_command(TARGET mysqld
+#     PRE_LINK
+#     COMMAND CreateExportList /tmp/libmariadbd.exp libsql.a ../mysys/libmysys.a ../dbug/libdbug.a ../strings/libstrings.a ../vio/libvio.a -X${OBJECT_MODE}
+#     VERBATIM
+#   )
+  # strings, mysys_ssl? pcre?
+ENDIF()
+
 IF(APPLE) 
   # Add CoreServices framework since some dloadable plugins may need it 
   FIND_LIBRARY(CORESERVICES NAMES CoreServices) 
--- ./mysys/CMakeLists.txt.orig	2019-06-27 11:46:34.000000000 +0200
+++ ./mysys/CMakeLists.txt	2019-06-27 11:46:34.000000000 +0200
@@ -92,6 +92,10 @@
 ENDIF()
 
 ADD_EXECUTABLE(thr_lock thr_lock.c)
+IF (AIX)
+  ADD_DEPENDENCIES(thr_lock mysqld)
+  TARGET_LINK_OPTIONS(thr_lock PRIVATE "-Wl,-bI:/tmp/libmariadbd.exp")
+ENDIF()
 TARGET_LINK_LIBRARIES(thr_lock mysys)
 SET_TARGET_PROPERTIES(thr_lock PROPERTIES COMPILE_FLAGS "-DMAIN")
 
