--- ./storage/innobase/include/univ.i.priu64	2020-08-07 14:57:37 +0200
+++ ./storage/innobase/include/univ.i	2020-09-11 14:12:21 +0200
@@ -404,6 +404,12 @@
 # define INT64PF	"%lld"
 # define UINT64scan     "llu"
 # define UINT64PFx	"%016llx"
+#elif defined _AIX
+/* Workaround for macros expension trouble */
+# define UINT32PF      "%u"
+# define INT64PF       "%lld"
+# define UINT64scan    "lu"
+# define UINT64PFx     "%016llx"
 #else
 /* Use the integer types and formatting strings defined in the C99 standard. */
 # define UINT32PF	"%" PRIu32
