--- ./include/ma_global.h-ORIGIN	2018-09-14 13:26:34 -0500
+++ ./include/ma_global.h	2018-09-14 13:27:11 -0500
@@ -615,7 +615,7 @@
 typedef char	*gptr;		/* Generic pointer */
 #endif
 #ifndef HAVE_INT_8_16_32
-typedef char	int8;		/* Signed integer >= 8	bits */
+typedef signed char	int8;		/* Signed integer >= 8	bits */
 typedef short	int16;		/* Signed integer >= 16 bits */
 #endif
 #ifndef HAVE_UCHAR
