Index: httpd-2.2.22/srclib/apr/build/apr_hints.m4
===================================================================
--- httpd-2.2.22.orig/srclib/apr/build/apr_hints.m4	2011-04-12 19:35:55.000000000 +0200
+++ httpd-2.2.22/srclib/apr/build/apr_hints.m4	2012-05-10 16:03:50.000000000 +0200
@@ -68,6 +68,9 @@
             *-ibm-aix5*)
 	        APR_ADDTO(CPPFLAGS, [-D_USE_IRS])
 	        ;;
+            *-ibm-aix6*)
+	        APR_ADDTO(CPPFLAGS, [-D_USE_IRS])
+	        ;;
             *-ibm-aix4.3.*)
                 APR_ADDTO(CPPFLAGS, [-D_USE_IRS])
                 ;;
Index: httpd-2.2.22/srclib/apr/configure.in
===================================================================
--- httpd-2.2.22.orig/srclib/apr/configure.in	2011-04-11 21:58:38.000000000 +0200
+++ httpd-2.2.22/srclib/apr/configure.in	2012-05-10 16:09:02.000000000 +0200
@@ -1710,7 +1710,7 @@
          pid_t_fmt='#define APR_PID_T_FMT "ld"'
        fi
        ;;
-   *aix4*|*aix5*)
+   *aix4*|*aix5*|aix6*)
        ssize_t_fmt="ld"
        size_t_fmt="lu"
        ;;
