--- a/src/modules/perl/modperl_perl.c	2022-01-13 08:47:15.720110701 +0000
+++ b/src/modules/perl/modperl_perl.c	2022-01-13 08:47:21.940180785 +0000
@@ -268,7 +268,7 @@
 #ifdef MP_NEED_HASH_SEED_FIXUP
     if (MP_init_hash_seed_set) {
 #if MP_PERL_VERSION_AT_LEAST(5, 17, 6)
-        memcpy(&PL_hash_seed, &MP_init_hash_seed,
+        memcpy(PL_hash_seed, &MP_init_hash_seed,
                 sizeof(PL_hash_seed) > sizeof(MP_init_hash_seed) ?
                     sizeof(MP_init_hash_seed) : sizeof(PL_hash_seed));
         PL_hash_seed_set   = MP_init_hash_seed_set;
