diff -crN ./gettext/install-sh /users3/olivap/sb/p64/src/./gettext/install-sh
*** ./gettext/install-sh	Sat Jan  6 01:56:08 2001
--- /users3/olivap/sb/p64/src/./gettext/install-sh	Thu Dec 22 17:21:01 2005
***************
*** 196,202 ****
  
  	if [ x"$chowncmd" != x ]; then $doit $chowncmd "$dst"; else true ; fi &&
  	if [ x"$chgrpcmd" != x ]; then $doit $chgrpcmd "$dst"; else true ; fi &&
! 	if [ x"$stripcmd" != x ]; then $doit $stripcmd "$dst"; else true ; fi &&
  	if [ x"$chmodcmd" != x ]; then $doit $chmodcmd "$dst"; else true ; fi
  else
  
--- 196,202 ----
  
  	if [ x"$chowncmd" != x ]; then $doit $chowncmd "$dst"; else true ; fi &&
  	if [ x"$chgrpcmd" != x ]; then $doit $chgrpcmd "$dst"; else true ; fi &&
! 	if [ x"$stripcmd" != x ]; then $doit $stripcmd "$dst" || true; else true ; fi &&
  	if [ x"$chmodcmd" != x ]; then $doit $chmodcmd "$dst"; else true ; fi
  else
  
***************
*** 241,247 ****
  
  	if [ x"$chowncmd" != x ]; then $doit $chowncmd "$dsttmp";else true;fi &&
  	if [ x"$chgrpcmd" != x ]; then $doit $chgrpcmd "$dsttmp";else true;fi &&
! 	if [ x"$stripcmd" != x ]; then $doit $stripcmd "$dsttmp";else true;fi &&
  	if [ x"$chmodcmd" != x ]; then $doit $chmodcmd "$dsttmp";else true;fi &&
  
  # Now remove or move any old file at destination location.  We try this
--- 241,247 ----
  
  	if [ x"$chowncmd" != x ]; then $doit $chowncmd "$dsttmp";else true;fi &&
  	if [ x"$chgrpcmd" != x ]; then $doit $chgrpcmd "$dsttmp";else true;fi &&
! 	if [ x"$stripcmd" != x ]; then $doit $stripcmd "$dsttmp" || true;else true;fi &&
  	if [ x"$chmodcmd" != x ]; then $doit $chmodcmd "$dsttmp";else true;fi &&
  
  # Now remove or move any old file at destination location.  We try this
