*** Xvnc/config/cf/vnc.def.ORIG	Wed Mar 11 21:30:13 2009
--- Xvnc/config/cf/vnc.def	Wed Mar 11 21:30:31 2009
***************
*** 108,114 ****
  
  #define XVendorString		\"AT&T Laboratories Cambridge\"
  #define XVendorRelease		3332
! #define XvncRelease		\"TightVNC-1.3.10\"
  
  
  /*****************************************************************************/
--- 108,114 ----
  
  #define XVendorString		\"AT&T Laboratories Cambridge\"
  #define XVendorRelease		3332
! #define XvncRelease		\""TightVNC-1.3.10\""
  
  
  /*****************************************************************************/
