Index: Python-2.7.2/Lib/cgi.py
===================================================================
--- Python-2.7.2.orig/Lib/cgi.py	2011-06-11 17:46:23.000000000 +0200
+++ Python-2.7.2/Lib/cgi.py	2011-10-05 10:35:22.000000000 +0200
@@ -1,4 +1,4 @@
-#! /usr/local/bin/python
+#! /opt/freeware/bin/python
 
 # NOTE: the above "/usr/local/bin/python" is NOT a mistake.  It is
 # intentionally NOT "/usr/bin/env python".  On many systems
Index: Python-2.7.2/Lib/distutils/sysconfig.py
===================================================================
--- Python-2.7.2.orig/Lib/distutils/sysconfig.py	2011-06-11 17:46:24.000000000 +0200
+++ Python-2.7.2/Lib/distutils/sysconfig.py	2011-10-05 10:18:36.000000000 +0200
@@ -403,7 +403,7 @@
 
     elif get_python_version() < '2.1':
         # The following two branches are for 1.5.2 compatibility.
-        if sys.platform == 'aix4':          # what about AIX 3.x ?
+        if sys.platform in [ 'aix4', 'aix5', 'aix6', 'aix7']:          # what about AIX 3.x ?
             # Linker script is in the config directory, not in Modules as the
             # Makefile says.
             python_lib = get_python_lib(standard_lib=1)
Index: Python-2.7.2/Lib/posixfile.py
===================================================================
--- Python-2.7.2.orig/Lib/posixfile.py	2011-06-11 17:46:25.000000000 +0200
+++ Python-2.7.2/Lib/posixfile.py	2011-10-05 10:24:44.000000000 +0200
@@ -185,7 +185,7 @@
                             'bsdos2', 'bsdos3', 'bsdos4'):
             flock = struct.pack('lxxxxlxxxxlhh', \
                   l_start, l_len, os.getpid(), l_type, l_whence)
-        elif sys.platform in ('aix3', 'aix4'):
+        elif sys.platform in ('aix3', 'aix4', 'aix5', 'aix6', 'aix7'):
             flock = struct.pack('hhlllii', \
                   l_type, l_whence, l_start, l_len, 0, 0, 0)
         else:
@@ -201,7 +201,7 @@
                                 'bsdos2', 'bsdos3', 'bsdos4'):
                 l_start, l_len, l_pid, l_type, l_whence = \
                     struct.unpack('lxxxxlxxxxlhh', flock)
-            elif sys.platform in ('aix3', 'aix4'):
+            elif sys.platform in ('aix3', 'aix4', 'aix5', 'aix6', 'aix7'):
                 l_type, l_whence, l_start, l_len, l_sysid, l_pid, l_vfs = \
                     struct.unpack('hhlllii', flock)
             elif sys.platform == "linux2":
Index: Python-2.7.2/Lib/test/test_fcntl.py
===================================================================
--- Python-2.7.2.orig/Lib/test/test_fcntl.py	2011-06-11 17:46:25.000000000 +0200
+++ Python-2.7.2/Lib/test/test_fcntl.py	2011-10-05 10:26:10.000000000 +0200
@@ -41,7 +41,7 @@
             pid_t = 'l'
         lockdata = struct.pack(off_t + off_t + pid_t + 'hh', 0, 0, 0,
                                fcntl.F_WRLCK, 0)
-    elif sys.platform in ['aix3', 'aix4', 'hp-uxB', 'unixware7']:
+    elif sys.platform in ['aix3', 'aix4', 'aix5', 'aix6', 'aix7', 'hp-uxB', 'unixware7']:
         lockdata = struct.pack('hhlllii', fcntl.F_WRLCK, 0, 0, 0, 0, 0, 0)
     elif sys.platform in ['os2emx']:
         lockdata = None
Index: Python-2.7.2/Makefile.pre.in
===================================================================
--- Python-2.7.2.orig/Makefile.pre.in	2011-06-11 17:46:26.000000000 +0200
+++ Python-2.7.2/Makefile.pre.in	2011-10-05 10:28:16.000000000 +0200
@@ -425,10 +425,10 @@
 
 libpython$(VERSION).so: $(LIBRARY_OBJS)
 	if test $(INSTSONAME) != $(LDLIBRARY); then \
-		$(BLDSHARED) -Wl,-h$(INSTSONAME) -o $(INSTSONAME) $(LIBRARY_OBJS) $(MODLIBS) $(SHLIBS) $(LIBC) $(LIBM) $(LDLAST); \
+		$(BLDSHARED) $(LDFLAGS) -Wl,-h$(INSTSONAME) -o $(INSTSONAME) $(LIBRARY_OBJS) $(MODLIBS) $(SHLIBS) $(LIBC) $(LIBM) $(LDLAST); \
 		$(LN) -f $(INSTSONAME) $@; \
 	else \
-		$(BLDSHARED) -o $@ $(LIBRARY_OBJS) $(MODLIBS) $(SHLIBS) $(LIBC) $(LIBM) $(LDLAST); \
+		$(BLDSHARED) $(LDFLAGS) -o $@ $(LIBRARY_OBJS) $(MODLIBS) $(SHLIBS) $(LIBC) $(LIBM) $(LDLAST); \
 	fi
 
 libpython$(VERSION).dylib: $(LIBRARY_OBJS)
Index: Python-2.7.2/Modules/Setup.dist
===================================================================
--- Python-2.7.2.orig/Modules/Setup.dist	2011-06-11 17:46:26.000000000 +0200
+++ Python-2.7.2/Modules/Setup.dist	2011-10-05 10:29:49.000000000 +0200
@@ -153,7 +153,7 @@
 # modules are to be built as shared libraries (see above for more
 # detail; also note that *static* reverses this effect):
 
-#*shared*
+*shared*
 
 # GNU readline.  Unlike previous Python incarnations, GNU readline is
 # now incorporated in an optional module, configured in the Setup file
Index: Python-2.7.2/configure
===================================================================
--- Python-2.7.2.orig/configure	2011-06-11 17:46:28.000000000 +0200
+++ Python-2.7.2/configure	2011-10-05 10:32:02.000000000 +0200
@@ -3084,7 +3084,7 @@
   # or has another value. By not (re)defining it, the defaults come in place.
   AIX/4)
     define_xopen_source=no;;
-  AIX/5)
+  AIX/5|AIX/6|AIX/7)
     if test `uname -r` -eq 1; then
       define_xopen_source=no
     fi
@@ -4965,7 +4965,7 @@
 	BLDLIBRARY='-L. -lpython$(VERSION)'
 	RUNSHARED='DYLD_LIBRARY_PATH=`pwd`:${DYLD_LIBRARY_PATH}'
 	;;
-    AIX*)
+    aix*|AIX*)
 	LDLIBRARY='libpython$(VERSION).so'
 	RUNSHARED=LIBPATH=`pwd`:${LIBPATH}
 	;;
Index: Python-2.7.2/configure.in
===================================================================
--- Python-2.7.2.orig/configure.in	2011-06-11 17:46:28.000000000 +0200
+++ Python-2.7.2/configure.in	2011-10-05 10:33:20.000000000 +0200
@@ -371,7 +371,7 @@
   # or has another value. By not (re)defining it, the defaults come in place.
   AIX/4)
     define_xopen_source=no;;
-  AIX/5)
+  AIX/5|AIX/6|AIX/7)
     if test `uname -r` -eq 1; then
       define_xopen_source=no
     fi
@@ -816,7 +816,7 @@
 	BLDLIBRARY='-L. -lpython$(VERSION)'
 	RUNSHARED='DYLD_LIBRARY_PATH=`pwd`:${DYLD_LIBRARY_PATH}'
 	;;
-    AIX*)
+    aix*|AIX*)
 	LDLIBRARY='libpython$(VERSION).so'
 	RUNSHARED=LIBPATH=`pwd`:${LIBPATH}
 	;;
Index: Python-2.7.2/setup.py
===================================================================
--- Python-2.7.2.orig/setup.py	2011-10-05 10:12:24.000000000 +0200
+++ Python-2.7.2/setup.py	2011-10-05 10:34:32.000000000 +0200
@@ -1794,7 +1794,7 @@
         libs.append('tk'+ version)
         libs.append('tcl'+ version)
 
-        if platform in ['aix3', 'aix4']:
+        if platform in ['aix3', 'aix4', 'aix5', 'aix6', 'aix7']:
             libs.append('ld')
 
         # Finally, link with the X11 libraries (not appropriate on cygwin)
