--- old/Python/thread_pthread.h	2016-07-28 16:36:45 +0200
+++ new/Python/thread_pthread.h	2016-07-28 16:36:53 +0200
@@ -12,6 +12,10 @@
 #endif
 #include <signal.h>
 
+#ifdef _AIX
+#define THREAD_STACK_SIZE       0x400000
+#endif
+
 /* The POSIX spec requires that use of pthread_attr_setstacksize
    be conditional on _POSIX_THREAD_ATTR_STACKSIZE being defined. */
 #ifdef _POSIX_THREAD_ATTR_STACKSIZE
