Index: Python-2.5.2/Modules/termios.c
===================================================================
--- Python-2.5.2.orig/Modules/termios.c	2006-01-19 07:09:39.000000000 +0100
+++ Python-2.5.2/Modules/termios.c	2008-07-31 16:56:43.-1048188840 +0200
@@ -11,6 +11,18 @@
 #endif
 
 #include <termios.h>
+
+#ifndef VWERASE
+#ifdef VWERSE
+	#define VWERASE VWERSE
+#endif
+#endif
+#ifndef VDISCARD
+	#ifdef VDISCRD
+	#define VDISCARD VDISCRD
+	#endif
+#endif
+
 #ifdef __osf__
 /* On OSF, sys/ioctl.h requires that struct termio already be defined,
  * so this needs to be included first on that platform. */
