--- ./Makefile.in	2020-08-20 11:46:14 +0200
+++ ./Makefile.in	2020-08-20 11:46:21 +0200
@@ -50,7 +50,7 @@
 libdir = ${exec_prefix}/lib
 sharedlibdir = ${libdir}
 includedir = ${prefix}/include
-mandir = ${prefix}/share/man
+mandir = ${prefix}/man
 man3dir = ${mandir}/man3
 pkgconfigdir = ${libdir}/pkgconfig
 SRCDIR=
--- ./configure	2020-08-20 11:47:08 +0200
+++ ./configure	2020-08-20 11:47:11 +0200
@@ -76,7 +76,7 @@
 libdir=${libdir-'${exec_prefix}/lib'}
 sharedlibdir=${sharedlibdir-'${libdir}'}
 includedir=${includedir-'${prefix}/include'}
-mandir=${mandir-'${prefix}/share/man'}
+mandir=${mandir-'${prefix}/man'}
 shared_ext='.so'
 shared=1
 solo=0
