--- ./Makefile.orig	2020-07-16 13:04:25 +0200
+++ ./Makefile	2020-07-16 13:04:47 +0200
@@ -204,7 +204,9 @@
 ifeq ($(CONFIG_TARGET_OS), AIX)
   LIBS	 += -lpthread -ldl -lrt
   CPPFLAGS += -D_LARGE_FILES -D__ppc__
-  LDFLAGS += -L/opt/freeware/lib -Wl,-blibpath:/opt/freeware/lib:/usr/lib:/lib -Wl,-bmaxdata:0x80000000
+  LDFLAGS += -L/opt/freeware/lib
+  SOURCE += oslib/strcasestr.c
+  T_ZIPF_OBJS += oslib/strcasestr.o
 endif
 ifeq ($(CONFIG_TARGET_OS), HP-UX)
   LIBS   += -lpthread -ldl -lrt
