*** ./Makefile.ORIG	Wed Apr 20 21:30:02 2016
--- ./Makefile	Wed Apr 20 21:30:22 2016
***************
*** 161,166 ****
--- 161,168 ----
    LIBS	 += -lpthread -ldl -lrt
    CPPFLAGS += -D_LARGE_FILES -D__ppc__
    LDFLAGS += -L/opt/freeware/lib -Wl,-blibpath:/opt/freeware/lib:/usr/lib:/lib -Wl,-bmaxdata:0x80000000
+   SOURCE += oslib/strcasestr.c
+   T_ZIPF_OBJS += oslib/strcasestr.o
  endif
  ifeq ($(CONFIG_TARGET_OS), HP-UX)
    LIBS   += -lpthread -ldl -lrt
