Index: lua-5.1.4/Makefile
===================================================================
--- lua-5.1.4.orig/Makefile	2008-08-12 02:40:48.000000000 +0200
+++ lua-5.1.4/Makefile	2009-02-10 18:32:50.248024088 +0100
@@ -9,7 +9,7 @@
 
 # Where to install. The installation starts in the src and doc directories,
 # so take care if INSTALL_TOP is not an absolute path.
-INSTALL_TOP= /usr/local
+INSTALL_TOP= $(DESTDIR)/opt/freeware
 INSTALL_BIN= $(INSTALL_TOP)/bin
 INSTALL_INC= $(INSTALL_TOP)/include
 INSTALL_LIB= $(INSTALL_TOP)/lib
@@ -22,14 +22,14 @@
 
 # How to install. If your install program does not support "-p", then you
 # may have to run ranlib on the installed liblua.a (do "make ranlib").
-INSTALL= install -p
-INSTALL_EXEC= $(INSTALL) -m 0755
-INSTALL_DATA= $(INSTALL) -m 0644
+# INSTALL= install -p
+# INSTALL_EXEC= $(INSTALL) -m 0755
+# INSTALL_DATA= $(INSTALL) -m 0644
 #
 # If you don't have install you can use cp instead.
-# INSTALL= cp -p
-# INSTALL_EXEC= $(INSTALL)
-# INSTALL_DATA= $(INSTALL)
+INSTALL= cp -p
+INSTALL_EXEC= $(INSTALL)
+INSTALL_DATA= $(INSTALL)
 
 # Utilities.
 MKDIR= mkdir -p
Index: lua-5.1.4/src/Makefile
===================================================================
--- lua-5.1.4.orig/src/Makefile	2008-01-19 20:37:58.000000000 +0100
+++ lua-5.1.4/src/Makefile	2009-02-10 18:06:06.679199512 +0100
@@ -5,17 +5,17 @@
 # == CHANGE THE SETTINGS BELOW TO SUIT YOUR ENVIRONMENT =======================
 
 # Your platform. See PLATS for possible values.
-PLAT= none
+PLAT= all
 
-CC= gcc
+CC= /usr/vacpp/bin/xlc_r
 CFLAGS= -O2 -Wall $(MYCFLAGS)
-AR= ar rcu
+AR= /usr/bin/ar -qv
 RANLIB= ranlib
 RM= rm -f
 LIBS= -lm $(MYLIBS)
 
 MYCFLAGS=
-MYLDFLAGS=
+MYLDFLAGS= -L/opt/freeware/lib
 MYLIBS=
 
 # == END OF USER SETTINGS. NO NEED TO CHANGE ANYTHING BELOW THIS LINE =========
@@ -81,7 +81,7 @@
 	@echo "   $(PLATS)"
 
 aix:
-	$(MAKE) all CC="xlc" CFLAGS="-O2 -DLUA_USE_POSIX -DLUA_USE_DLOPEN" MYLIBS="-ldl" MYLDFLAGS="-brtl -bexpall"
+	$(MAKE) all CC=$(CC) CFLAGS="-O2 -DLUA_USE_POSIX -DLUA_USE_DLOPEN" MYLIBS="-ldl" MYLDFLAGS="-brtl -bexpall"
 
 ansi:
 	$(MAKE) all MYCFLAGS=-DLUA_ANSI
