--- ./src/config/shlib.conf.aix-test-config	2020-02-12 18:21:58 +0100
+++ ./src/config/shlib.conf	2020-03-11 15:58:39 +0100
@@ -472,7 +472,7 @@
 	PROFFLAGS=-pg
 	;;
 
-*-*-aix5*)
+*-*-aix5*|*-*-aix6*|*-*-aix7*)
 	SHLIBVEXT='.so.$(LIBMAJOR).$(LIBMINOR)'
 	SHLIBEXT=.so
 	# AIX doesn't need separate PIC objects
@@ -489,9 +489,8 @@
 	  LDCOMBINE='/bin/ld -o $@ $$objlist -H512 -T512 -bnoentry -bgcbypass:1 -bnodelcsect -bfilelist -bM:SRE -bE:$(SHLIB_EXPORT_FILE) -bernotok -brtl $(SHLIB_EXPFLAGS) $(LDFLAGS) -lc $$initfini'
 	fi
 	# Assume initialization always delayed.
-	INIT_FINI_PREP="wl=${wl_prefix}; "'i=1; initfini=; for f in . $(LIBFINIFUNC); do if test $$f != .; then initfini="$$initfini $${wl}-binitfini::$$f:$$i"; else :; fi; i=`expr $$i + 1`; done'
 	use_linker_fini_option=yes
-	MAKE_SHLIB_COMMAND="${INIT_FINI_PREP} && ${LDCOMBINE}"
+	MAKE_SHLIB_COMMAND="${LDCOMBINE}"
 	RPATH_TAIL=:/usr/lib:/lib
 	PROG_RPATH_FLAGS='$(RPATH_FLAG)$(PROG_RPATH):'"$RPATH_TAIL"
 	CC_LINK_SHARED='$(CC) $(PROG_LIBPATH) $(PROG_RPATH_FLAGS) $(CFLAGS) $(LDFLAGS)'
@@ -499,7 +498,7 @@
 	CXX_LINK_SHARED='$(CXX) $(PROG_LIBPATH) $(PROG_RPATH_FLAGS) $(CXXFLAGS) $(LDFLAGS)'
 	CXX_LINK_STATIC='$(CXX) $(PROG_LIBPATH) $(CXXFLAGS) $(LDFLAGS)'
 	# $(PROG_RPATH) is here to handle things like a shared tcl library
-	RUN_ENV='LIBPATH=`echo $(PROG_LIBPATH) | sed -e "s/-L//g" -e "s/ /:/g"`:$(PROG_RPATH):/usr/lib:/usr/local/lib'
+	RUN_ENV='LIBPATH=`echo $(PROG_LIBPATH) | sed -e "s/-L//g" -e "s/ /:/g"`:$(PROG_RPATH):/opt/freeware/lib:/usr/lib:/usr/local/lib'
 	RUN_VARS='LIBPATH'
 	;;
 
