Index: vim73/src/os_unix.c
===================================================================
--- vim73.orig/src/os_unix.c	2012-04-11 13:15:09.000000000 +0200
+++ vim73/src/os_unix.c	2012-04-11 13:24:27.000000000 +0200
@@ -1889,6 +1889,9 @@
 }
 
 /* Are Xutf8 functions available?  Avoid error from old compilers. */
+#if defined(X_HAVE_UTF8_STRING)
+# undef X_HAVE_UTF8_STRING
+#endif
 #if defined(X_HAVE_UTF8_STRING) && defined(FEAT_MBYTE)
 # if X_HAVE_UTF8_STRING
 #  define USE_UTF8_STRING
