From 7faa23a0e04889b627265d8acaaa3fdbfa92f520 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Cl=C3=A9ment=20Chigot?= <clement.chigot@atos.net>
Date: Wed, 15 Sep 2021 16:23:28 +0200
Subject: [PATCH] lib: remove err.h if not available

---
 lib/pager.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/pager.c b/lib/pager.c
index 747521e..7e811e2 100644
--- a/lib/pager.c
+++ b/lib/pager.c
@@ -8,7 +8,9 @@
 #include <unistd.h>
 #include <stdlib.h>
 #include <string.h>
+#ifdef HAVE_ERR_H
 #include <err.h>
+#endif
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/wait.h>
-- 
2.33.0

