Index: lzma-4.32.7/config.guess
===================================================================
--- lzma-4.32.7.orig/config.guess	2007-11-06 20:33:53.000000000 +0100
+++ lzma-4.32.7/config.guess	2012-09-06 07:54:39.000000000 +0200
@@ -531,7 +531,7 @@
 		echo rs6000-ibm-aix3.2
 	fi
 	exit ;;
-    *:AIX:*:[45])
+    *:AIX:*:[456])
 	IBM_CPU_ID=`/usr/sbin/lsdev -C -c processor -S available | sed 1q | awk '{ print $1 }'`
 	if /usr/sbin/lsattr -El ${IBM_CPU_ID} | grep ' POWER' >/dev/null 2>&1; then
 		IBM_ARCH=rs6000
