diff -crN ./Xrender/configure.ac /gestconf/project/GNOME_ACL/GNOME/build/sh_build_GNOME/src/./Xrender/configure.ac
*** ./Xrender/configure.ac	Thu Jan 15 09:03:19 2004
--- /gestconf/project/GNOME_ACL/GNOME/build/sh_build_GNOME/src/./Xrender/configure.ac	Wed Oct 12 09:44:45 2005
***************
*** 1,5 ****
  dnl 
! dnl  $Id: configure.ac,v 1.18 2004/01/15 08:03:19 daniel Exp $
  dnl 
  dnl  Copyright © 2003 Keith Packard, Noah Levitt
  dnl 
--- 1,5 ----
  dnl 
! dnl  $Id: configure.ac,v 1.1.3.3 2005/10/12 07:44:43 bergerv Exp $
  dnl 
  dnl  Copyright © 2003 Keith Packard, Noah Levitt
  dnl 
***************
*** 23,29 ****
  dnl
  dnl Process this file with autoconf to create configure.
  
! AC_PREREQ([2.57])
  
  dnl
  dnl Version should match the current Render version. XRenderQueryVersion
--- 23,29 ----
  dnl
  dnl Process this file with autoconf to create configure.
  
! AC_PREREQ([2.54])
  
  dnl
  dnl Version should match the current Render version. XRenderQueryVersion
***************
*** 33,43 ****
  dnl protocol, so Xrender version l.n.m corresponds to protocol version l.n
  dnl
  AC_INIT(libXrender, 0.8.4, [keithp@keithp.com], libXrender)
  AM_INIT_AUTOMAKE([dist-bzip2])
  AM_MAINTAINER_MODE
  
  AM_CONFIG_HEADER(config.h)
- AC_CONFIG_AUX_DIR(.)
  
  # Check for progs
  AC_PROG_CC
--- 33,43 ----
  dnl protocol, so Xrender version l.n.m corresponds to protocol version l.n
  dnl
  AC_INIT(libXrender, 0.8.4, [keithp@keithp.com], libXrender)
+ AC_CONFIG_AUX_DIR(.)
  AM_INIT_AUTOMAKE([dist-bzip2])
  AM_MAINTAINER_MODE
  
  AM_CONFIG_HEADER(config.h)
  
  # Check for progs
  AC_PROG_CC
