Index: gtk+-2.20.1/gdk-pixbuf/io-png.c
===================================================================
--- gtk+-2.20.1.orig/gdk-pixbuf/io-png.c	2010-05-02 05:14:27.000000000 +0200
+++ gtk+-2.20.1/gdk-pixbuf/io-png.c	2013-03-29 14:52:49.000000000 +0100
@@ -184,7 +184,12 @@
                              error_msg);
         }
 
-        longjmp (png_save_ptr->jmpbuf, 1);
+       /* TBD: libpng doesn't provide a way to directly invoke the
+       * stored 'longjmp', so do it by hand:
+       */
+       longjmp (png_jmpbuf(png_save_ptr), 1);
+
+
 }
 
 static void
@@ -288,7 +293,7 @@
 		return NULL;
 	}
 
-	if (setjmp (png_ptr->jmpbuf)) {
+       if (setjmp (png_jmpbuf(png_ptr))) {
 	    	g_free (rows);
 
 		if (pixbuf)
@@ -460,7 +465,7 @@
                 return NULL;
         }
         
-	if (setjmp (lc->png_read_ptr->jmpbuf)) {
+       if (setjmp (png_jmpbuf(lc->png_read_ptr))) {
 		if (lc->png_info_ptr)
 			png_destroy_read_struct(&lc->png_read_ptr, NULL, NULL);
                 g_free(lc);
@@ -532,7 +537,7 @@
         lc->error = error;
         
         /* Invokes our callbacks as needed */
-	if (setjmp (lc->png_read_ptr->jmpbuf)) {
+       if (setjmp (png_jmpbuf(lc->png_read_ptr))) {
                 lc->error = NULL;
 		return FALSE;
 	} else {
@@ -654,7 +659,7 @@
                         g_set_error (lc->error,
                                      GDK_PIXBUF_ERROR,
                                      GDK_PIXBUF_ERROR_INSUFFICIENT_MEMORY,
-                                     _("Insufficient memory to store a %ld by %ld image; try exiting some applications to reduce memory usage"),
+                                    _("Insufficient memory to store a %u by %u image; try exiting some applications to reduce memory usage"),
                                      width, height);
                 }
                 return;
@@ -770,7 +775,12 @@
                              error_msg);
         }
 
-        longjmp (png_read_ptr->jmpbuf, 1);
+       /* TBD: libpng doesn't provide a way to directly invoke the
+       * stored 'longjmp', so do it by hand:
+       */
+       longjmp (png_jmpbuf(png_read_ptr), 1);
+
+
 }
 
 static void
@@ -979,7 +989,7 @@
 	       success = FALSE;
 	       goto cleanup;
        }
-       if (setjmp (png_ptr->jmpbuf)) {
+       if (setjmp (png_jmpbuf(png_ptr))) {
 	       success = FALSE;
 	       goto cleanup;
        }
